<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Allele-specific transcriptomics tutorial</title>
  <meta name="description" content="Allele-specific transcriptomics tutorial" />
  <meta name="generator" content="bookdown 0.38 and GitBook 2.6.7" />

  <meta property="og:title" content="Allele-specific transcriptomics tutorial" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Allele-specific transcriptomics tutorial" />
  
  
  

<meta name="author" content="Sean T. Bresnahan" />


<meta name="date" content="2024-07-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">AST Tutorial</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#contents"><i class="fa fa-check"></i><b>1.1</b> Contents</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#sequence-reads-to-snp-level-parent-specific-read-counts"><i class="fa fa-check"></i>Sequence reads to SNP-level parent-specific read counts</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#snp-level-parent-specific-read-counts-to-allele-specific-transcription"><i class="fa fa-check"></i>SNP-level parent-specific read counts to allele-specific transcription</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#data"><i class="fa fa-check"></i><b>1.2</b> Data</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><i class="fa fa-check"></i><b>2</b> Sequence reads to SNP-level parent-specific read counts</a>
<ul>
<li class="chapter" data-level="2.1" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#overview"><i class="fa fa-check"></i><b>2.1</b> Overview</a></li>
<li class="chapter" data-level="2.2" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#setup"><i class="fa fa-check"></i><b>2.2</b> Setup</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#command-line-tools"><i class="fa fa-check"></i><b>2.2.1</b> Command-line tools</a></li>
<li class="chapter" data-level="" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#project-directory-structure"><i class="fa fa-check"></i>Project directory structure</a></li>
<li class="chapter" data-level="2.2.2" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#amel_hav3.1-reference-genome-assembly-and-annotation-files"><i class="fa fa-check"></i><b>2.2.2</b> Amel_HAv3.1 reference genome assembly and annotation files</a></li>
<li class="chapter" data-level="2.2.3" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#a-note-on-parallel-processing-in-high-performance-computing-clusters"><i class="fa fa-check"></i><b>2.2.3</b> A note on parallel processing in high performance computing clusters</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#generation-of-f1-genomes-and-transcriptomes"><i class="fa fa-check"></i><b>2.3</b> Generation of F1 genomes and transcriptomes</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#wgs-read-pre-processing"><i class="fa fa-check"></i><b>2.3.1</b> WGS read pre-processing</a></li>
<li class="chapter" data-level="2.3.2" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#wgs-read-alignment-and-post-processing"><i class="fa fa-check"></i><b>2.3.2</b> WGS read alignment and post-processing</a></li>
<li class="chapter" data-level="2.3.3" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#variant-discovery-and-filtration"><i class="fa fa-check"></i><b>2.3.3</b> Variant discovery and filtration</a></li>
<li class="chapter" data-level="2.3.4" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#genome-and-transcriptome-construction"><i class="fa fa-check"></i><b>2.3.4</b> Genome and transcriptome construction</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#quantification-of-f2-allele-specific-transcription"><i class="fa fa-check"></i><b>2.4</b> Quantification of F2 allele-specific transcription</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#mrna-seq-read-pre-processing"><i class="fa fa-check"></i><b>2.4.1</b> mRNA-seq read pre-processing</a></li>
<li class="chapter" data-level="2.4.2" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#mrna-seq-read-alignment-to-f1-transcriptomes"><i class="fa fa-check"></i><b>2.4.2</b> mRNA-seq read alignment to F1 transcriptomes</a></li>
<li class="chapter" data-level="2.4.3" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#mrna-seq-alignment-post-processing-and-allele-specific-read-counting"><i class="fa fa-check"></i><b>2.4.3</b> mRNA-seq alignment post-processing and allele-specific read counting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><i class="fa fa-check"></i><b>3</b> SNP-level parent-specific read counts to allele-specific transcription</a>
<ul>
<li class="chapter" data-level="3.1" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#overview-1"><i class="fa fa-check"></i><b>3.1</b> Overview</a></li>
<li class="chapter" data-level="3.2" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#setup-1"><i class="fa fa-check"></i><b>3.2</b> Setup</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#tutorial-files"><i class="fa fa-check"></i><b>3.2.1</b> Tutorial files</a></li>
<li class="chapter" data-level="3.2.2" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#packages"><i class="fa fa-check"></i><b>3.2.2</b> Packages</a></li>
<li class="chapter" data-level="3.2.3" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#custom-functions"><i class="fa fa-check"></i><b>3.2.3</b> Custom functions</a></li>
<li class="chapter" data-level="3.2.4" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#sample-metadata"><i class="fa fa-check"></i><b>3.2.4</b> Sample metadata</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#generate-sample-x-snpgene-count-matrix"><i class="fa fa-check"></i><b>3.3</b> Generate sample x SNP:gene count matrix</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#filter-snps-by-transcript"><i class="fa fa-check"></i><b>3.3.1</b> Filter SNPs by transcript</a></li>
<li class="chapter" data-level="3.3.2" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#merge-count-files-to-matrix"><i class="fa fa-check"></i><b>3.3.2</b> Merge count files to matrix</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#prepare-sample-x-snpgene-counts-for-analysis"><i class="fa fa-check"></i><b>3.4</b> Prepare sample x SNP:gene counts for analysis</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#normalize-by-library-size"><i class="fa fa-check"></i><b>3.4.1</b> Normalize by library size</a></li>
<li class="chapter" data-level="3.4.2" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#split-by-phenotype-and-filter-low-count-snps"><i class="fa fa-check"></i><b>3.4.2</b> Split by phenotype and filter low count SNPs</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#conduct-statistical-tests"><i class="fa fa-check"></i><b>3.5</b> Conduct statistical tests</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#snp-level-storer-kim-sk-binomial-exact-tests"><i class="fa fa-check"></i><b>3.5.1</b> SNP-level Storer-Kim (SK) binomial exact tests</a></li>
<li class="chapter" data-level="3.5.2" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#gene-level-general-linear-mixed-models-glimmix"><i class="fa fa-check"></i><b>3.5.2</b> Gene-level general linear mixed models (GLIMMIX)</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#allele-specific-transcription-analysis"><i class="fa fa-check"></i><b>3.6</b> Allele-specific transcription analysis</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#fdr-correction-and-threshold-tests"><i class="fa fa-check"></i><b>3.6.1</b> FDR correction and threshold tests</a></li>
<li class="chapter" data-level="3.6.2" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#chi-squared-tests"><i class="fa fa-check"></i><b>3.6.2</b> Chi-squared tests</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#data-visualization"><i class="fa fa-check"></i><b>3.7</b> Data visualization</a></li>
<li class="chapter" data-level="3.8" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#session-info"><i class="fa fa-check"></i><b>3.8</b> Session info</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Allele-specific transcriptomics tutorial</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Allele-specific transcriptomics tutorial</h1>
<p class="author"><em>Sean T. Bresnahan</em></p>
<p class="date"><em>2024-07-15</em></p>
</div>
<div id="introduction" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">1</span> Introduction<a href="index.html#introduction" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This tutorial demonstrates best practices for processing whole genome sequence (WGS) and mRNA sequencing (mRNA-seq) reads for allele-specific transcriptomics analysis. <br /></p>
<div id="contents" class="section level2 hasAnchor" number="1.1">
<h2><span class="header-section-number">1.1</span> Contents<a href="index.html#contents" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="sequence-reads-to-snp-level-parent-specific-read-counts" class="section level3 unnumbered hasAnchor">
<h3><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html">Sequence reads to SNP-level parent-specific read counts</a><a href="index.html#sequence-reads-to-snp-level-parent-specific-read-counts" class="anchor-section" aria-label="Anchor link to header"></a></h3>
</div>
<div id="snp-level-parent-specific-read-counts-to-allele-specific-transcription" class="section level3 unnumbered hasAnchor">
<h3><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html">SNP-level parent-specific read counts to allele-specific transcription</a><a href="index.html#snp-level-parent-specific-read-counts-to-allele-specific-transcription" class="anchor-section" aria-label="Anchor link to header"></a></h3>
</div>
</div>
<div id="data" class="section level2 hasAnchor" number="1.2">
<h2><span class="header-section-number">1.2</span> Data<a href="index.html#data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Sequencing data for this tutorial are described in <a href="https://doi.org/10.1101/2024.06.09.598129">Bresnahan <em>et al.</em>, 2024, “Intragenomic conflict underlies extreme phenotypic plasticity in queen-worker caste determination in honey bees (<em>Apis mellifera</em>)”, <em>bioRxiv</em></a>. This study used instrumental insemination to create reciprocal crosses between F1 honey bees derived from different F0 genetic stocks (referred to as “lineage A” and “lineage B”) in two distinct genetic Blocks, and manipulated the F2 larvae resulting from these crosses to induce development of either the worker or queen caste fate. Whole-genome sequencing (WGS) of genomic DNA isolated from the F1 males and females used to make these crosses, and mRNA-seq of RNA isolated from the F2 larvae resulting from these crosses, was performed for quantifying and comparing allele-specific transcription between worker and queen-destined larvae. <br /></p>
<p>In this tutorial, sequencing data from one genetic Block of these crosses (<em>n</em>=4 150x150bp paired-end WGS libraries and <em>n</em>=20 50x50bp paired-end stranded mRNA-seq libraries) will be used in order to reproduce <strong>Figure 4</strong> from the study.</p>
<div class="float">
<img src="AST-tutorial/fig4.png" alt="Figure 4. Queen-destined larvae show enriched paternal allele-biased transcription relative to worker-destined larvae. Allele-specific transcriptomes were assessed in F2 worker-destined larvae (WL) and queen-destined larvae (QL) collected from a reciprocal cross between different stocks of European honey bees. The x-axis represents, for each transcript, the proportion of lineage A reads in larvae with a lineage B mother and lineage A father (p1). The y-axis represents, for each transcript, the proportion of lineage A reads in larvae with a lineage A mother and lineage B father (p2). Each color represents a transcript which is significantly biased at all tested SNP positions: black is maternal (mat), green is lineage A, gold is lineage B, blue is paternal (pat), and grey is not significant. Center table: the number of transcripts showing each category of allelic bias and p-values for Chi-squared tests of independence for comparisons between the castes are indicated (NS = not significant). Significance of allele-biased transcription was determined using the overlap between two statistical tests: a general linear mixed model (GLIMMIX), and a Storer-Kim binomial exact test along with thresholds of p1&lt;0.4 and p2&gt;0.6 for maternal bias, p1&gt;0.6 and p2&lt;0.4 for paternal bias, p1&lt;0.4 and p2&lt;0.4 for lineage B bias and p1&gt;0.6 and p2&gt;0.6 for lineage A bias." />
<div class="figcaption"><strong>Figure 4.</strong> Queen-destined larvae show enriched paternal allele-biased transcription relative to worker-destined larvae. Allele-specific transcriptomes were assessed in F2 worker-destined larvae (WL) and queen-destined larvae (QL) collected from a reciprocal cross between different stocks of European honey bees. The x-axis represents, for each transcript, the proportion of lineage A reads in larvae with a lineage B mother and lineage A father (<em>p1</em>). The y-axis represents, for each transcript, the proportion of lineage A reads in larvae with a lineage A mother and lineage B father (<em>p2</em>). Each color represents a transcript which is significantly biased at all tested SNP positions: black is maternal (mat), green is lineage A, gold is lineage B, blue is paternal (pat), and grey is not significant. <strong>Center table</strong>: the number of transcripts showing each category of allelic bias and <em>p</em>-values for Chi-squared tests of independence for comparisons between the castes are indicated (NS = not significant). Significance of allele-biased transcription was determined using the overlap between two statistical tests: a general linear mixed model (GLIMMIX), and a Storer-Kim binomial exact test along with thresholds of <em>p1</em>&lt;0.4 and <em>p2</em>&gt;0.6 for maternal bias, <em>p1</em>&gt;0.6 and <em>p2</em>&lt;0.4 for paternal bias, <em>p1</em>&lt;0.4 and <em>p2</em>&lt;0.4 for lineage B bias and <em>p1</em>&gt;0.6 and <em>p2</em>&gt;0.6 for lineage A bias.</div>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/YOUR GITHUB USERNAME/YOUR REPO NAME/edit/main/index.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/YOUR GITHUB USERNAME/YOUR REPO NAME/blob/main/index.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
