<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3 SNP-level parent-specific read counts to allele-specific transcription | Allele-specific transcriptomics tutorial</title>
  <meta name="description" content="3 SNP-level parent-specific read counts to allele-specific transcription | Allele-specific transcriptomics tutorial" />
  <meta name="generator" content="bookdown 0.38 and GitBook 2.6.7" />

  <meta property="og:title" content="3 SNP-level parent-specific read counts to allele-specific transcription | Allele-specific transcriptomics tutorial" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3 SNP-level parent-specific read counts to allele-specific transcription | Allele-specific transcriptomics tutorial" />
  
  
  

<meta name="author" content="Sean T. Bresnahan" />


<meta name="date" content="2024-07-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">AST Tutorial</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#contents"><i class="fa fa-check"></i><b>1.1</b> Contents</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#sequence-reads-to-snp-level-parent-specific-read-counts"><i class="fa fa-check"></i>Sequence reads to SNP-level parent-specific read counts</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#snp-level-parent-specific-read-counts-to-allele-specific-transcription"><i class="fa fa-check"></i>SNP-level parent-specific read counts to allele-specific transcription</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#data"><i class="fa fa-check"></i><b>1.2</b> Data</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><i class="fa fa-check"></i><b>2</b> Sequence reads to SNP-level parent-specific read counts</a>
<ul>
<li class="chapter" data-level="2.1" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#overview"><i class="fa fa-check"></i><b>2.1</b> Overview</a></li>
<li class="chapter" data-level="2.2" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#setup"><i class="fa fa-check"></i><b>2.2</b> Setup</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#command-line-tools"><i class="fa fa-check"></i><b>2.2.1</b> Command-line tools</a></li>
<li class="chapter" data-level="" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#project-directory-structure"><i class="fa fa-check"></i>Project directory structure</a></li>
<li class="chapter" data-level="2.2.2" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#amel_hav3.1-reference-genome-assembly-and-annotation-files"><i class="fa fa-check"></i><b>2.2.2</b> Amel_HAv3.1 reference genome assembly and annotation files</a></li>
<li class="chapter" data-level="2.2.3" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#a-note-on-parallel-processing-in-high-performance-computing-clusters"><i class="fa fa-check"></i><b>2.2.3</b> A note on parallel processing in high performance computing clusters</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#generation-of-f1-genomes-and-transcriptomes"><i class="fa fa-check"></i><b>2.3</b> Generation of F1 genomes and transcriptomes</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#wgs-read-pre-processing"><i class="fa fa-check"></i><b>2.3.1</b> WGS read pre-processing</a></li>
<li class="chapter" data-level="2.3.2" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#wgs-read-alignment-and-post-processing"><i class="fa fa-check"></i><b>2.3.2</b> WGS read alignment and post-processing</a></li>
<li class="chapter" data-level="2.3.3" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#variant-discovery-and-filtration"><i class="fa fa-check"></i><b>2.3.3</b> Variant discovery and filtration</a></li>
<li class="chapter" data-level="2.3.4" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#genome-and-transcriptome-construction"><i class="fa fa-check"></i><b>2.3.4</b> Genome and transcriptome construction</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#quantification-of-f2-allele-specific-transcription"><i class="fa fa-check"></i><b>2.4</b> Quantification of F2 allele-specific transcription</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#mrna-seq-read-pre-processing"><i class="fa fa-check"></i><b>2.4.1</b> mRNA-seq read pre-processing</a></li>
<li class="chapter" data-level="2.4.2" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#mrna-seq-read-alignment-to-f1-transcriptomes"><i class="fa fa-check"></i><b>2.4.2</b> mRNA-seq read alignment to F1 transcriptomes</a></li>
<li class="chapter" data-level="2.4.3" data-path="sequence-reads-to-snp-level-parent-specific-read-counts-1.html"><a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html#mrna-seq-alignment-post-processing-and-allele-specific-read-counting"><i class="fa fa-check"></i><b>2.4.3</b> mRNA-seq alignment post-processing and allele-specific read counting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><i class="fa fa-check"></i><b>3</b> SNP-level parent-specific read counts to allele-specific transcription</a>
<ul>
<li class="chapter" data-level="3.1" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#overview-1"><i class="fa fa-check"></i><b>3.1</b> Overview</a></li>
<li class="chapter" data-level="3.2" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#setup-1"><i class="fa fa-check"></i><b>3.2</b> Setup</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#tutorial-files"><i class="fa fa-check"></i><b>3.2.1</b> Tutorial files</a></li>
<li class="chapter" data-level="3.2.2" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#packages"><i class="fa fa-check"></i><b>3.2.2</b> Packages</a></li>
<li class="chapter" data-level="3.2.3" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#custom-functions"><i class="fa fa-check"></i><b>3.2.3</b> Custom functions</a></li>
<li class="chapter" data-level="3.2.4" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#sample-metadata"><i class="fa fa-check"></i><b>3.2.4</b> Sample metadata</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#generate-sample-x-snpgene-count-matrix"><i class="fa fa-check"></i><b>3.3</b> Generate sample x SNP:gene count matrix</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#filter-snps-by-transcript"><i class="fa fa-check"></i><b>3.3.1</b> Filter SNPs by transcript</a></li>
<li class="chapter" data-level="3.3.2" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#merge-count-files-to-matrix"><i class="fa fa-check"></i><b>3.3.2</b> Merge count files to matrix</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#prepare-sample-x-snpgene-counts-for-analysis"><i class="fa fa-check"></i><b>3.4</b> Prepare sample x SNP:gene counts for analysis</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#normalize-by-library-size"><i class="fa fa-check"></i><b>3.4.1</b> Normalize by library size</a></li>
<li class="chapter" data-level="3.4.2" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#split-by-phenotype-and-filter-low-count-snps"><i class="fa fa-check"></i><b>3.4.2</b> Split by phenotype and filter low count SNPs</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#conduct-statistical-tests"><i class="fa fa-check"></i><b>3.5</b> Conduct statistical tests</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#snp-level-storer-kim-sk-binomial-exact-tests"><i class="fa fa-check"></i><b>3.5.1</b> SNP-level Storer-Kim (SK) binomial exact tests</a></li>
<li class="chapter" data-level="3.5.2" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#gene-level-general-linear-mixed-models-glimmix"><i class="fa fa-check"></i><b>3.5.2</b> Gene-level general linear mixed models (GLIMMIX)</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#allele-specific-transcription-analysis"><i class="fa fa-check"></i><b>3.6</b> Allele-specific transcription analysis</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#fdr-correction-and-threshold-tests"><i class="fa fa-check"></i><b>3.6.1</b> FDR correction and threshold tests</a></li>
<li class="chapter" data-level="3.6.2" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#chi-squared-tests"><i class="fa fa-check"></i><b>3.6.2</b> Chi-squared tests</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#data-visualization"><i class="fa fa-check"></i><b>3.7</b> Data visualization</a></li>
<li class="chapter" data-level="3.8" data-path="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#session-info"><i class="fa fa-check"></i><b>3.8</b> Session info</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Allele-specific transcriptomics tutorial</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">3</span> SNP-level parent-specific read counts to allele-specific transcription<a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#snp-level-parent-specific-read-counts-to-allele-specific-transcription-1" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="overview-1" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Overview<a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#overview-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This R tutorial follows <a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html">quantification of SNP-level parent-specific mRNA-seq read counts</a> and documents a statistical method for comparing allele-specific transcription between phenotypes in offspring derived from reciprocal crosses.</p>
<p>Sequencing data for this tutorial are described in <a href="https://doi.org/10.1101/2024.06.09.598129">Bresnahan <em>et al.</em>, 2024, “Intragenomic conflict underlies extreme phenotypic plasticity in queen-worker caste determination in honey bees (<em>Apis mellifera</em>)”, <em>bioRxiv</em></a>. See the <a href="index.html">tutorial README</a> for more details.</p>
<p>The honey bee gene annotations and the F1 SNPs intersecting honey bee genes generated in Section 2 are imported. SNPs are assigned to the longest transcript for each gene. SNPs intersecting <span class="math inline">\(n &gt; 2\)</span> genes or <span class="math inline">\(n = 2\)</span> genes on the same strand are discarded, as it is impossible to assign SNP-level counts to genes with transcripts that overlap on the same strand. Additionally, SNPs within miRNA, tRNA, and repetitive pseudogenes are discarded as these are inappropriate for allele-specific analysis using short sequencing reads (for details, see: <a href="https://doi.org/10.1038/hdy.2014.18">Wang &amp; Clark, 2014, <em>Heredity</em></a>.)</p>
<p>The F2 read counts at F1 SNPs generated in Section 2 are imported. SNPs with <span class="math inline">\(n &lt; 1\)</span> count in any Lineage are discarded. In cases where the distance between SNPs is shorter than the average read length (and thus the read counts are exactly the same for both SNPs), one SNP is chosen at random. Finally, genes with <span class="math inline">\(n &lt; 2\)</span> SNPs are discarded, as this method requires at least two independent observations per gene for statistical analysis.</p>
<p>To adjust for differences in sequencing depth between libraries, library size factors are estimated and the read counts are normalized using the median of ratios normalization (MRN) method from DESeq2.</p>
<p>For each SNP, a <a href="https://doi.org/10.1080/01621459.1990.10475318">Storer-Kim binomial exact test of two proportions</a> is conducted using the MRN counts to test the hypothesis that the proportion of maternal and paternal read counts are statistically different.</p>
<p>A general linear mixed-effects model with interaction terms (GLIMMIX) is fit for each gene to assess the effects of Parent, Lineage, and their interaction on the raw read counts at each SNP, using the log of the library size factors as an offset to adjust for variation in sequencing depth between libraries. A Wald test is performed to assess the statistical significance of the effects.</p>
<p>For a gene to be considered as showing parent-of-origin or lineage-of-origin effects, all SNPs are required to exhibit the same directional bias in the Storer-Kim test, and a strictly parent-of-origin or lineage-of-origin effect in the GLIMMIX. To avoid identifying genes with parent-of-origin effects influenced by lineage-of-origin effects, or lineage-of-origin genes influenced by parent-of-origin effects, genes with a significant interaction effect are considered unbiased. Additionally, the proportions of Lineage A reads in samples with a Lineage B mother and Lineage A father (proportion 1, or <em>p1</em>), and the proportion of Lineage A reads in samples with a Lineage A mother and Lineage B father (<em>p2</em>), are subjected to a threshold test following methods described in <a href="https://doi.org/10.1038/hdy.2014.18">Wang &amp; Clark, 2014, <em>Heredity</em></a>.</p>
</div>
<div id="setup-1" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Setup<a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#setup-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="tutorial-files" class="section level3 hasAnchor" number="3.2.1">
<h3><span class="header-section-number">3.2.1</span> Tutorial files<a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#tutorial-files" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Download all files used in this tutorial <a href="AST-tutorial.zip">here</a> or via the command line:</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb39-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb39-1" tabindex="-1"></a><span class="fu">wget</span> https://github.com/sbresnahan/AST-tutorial/blob/d99afee8a9b735b8ff927a921cf75ec2997fe322/AST-tutorial.zip</span></code></pre></div>
</div>
<div id="packages" class="section level3 hasAnchor" number="3.2.2">
<h3><span class="header-section-number">3.2.2</span> Packages<a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#packages" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Several CRAN packages are required for this tutorial. Use <code>pacman</code> to install those that are not already on your machine.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb40-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;pacman&quot;</span>)) <span class="fu">install.packages</span>(<span class="st">&quot;pacman&quot;</span>)</span>
<span id="cb40-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb40-2" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(<span class="st">&quot;tidyverse&quot;</span>, <span class="st">&quot;plyr&quot;</span>, <span class="st">&quot;Rfast&quot;</span>, <span class="st">&quot;tryCatchLog&quot;</span>, </span>
<span id="cb40-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb40-3" tabindex="-1"></a>               <span class="st">&quot;lmerTest&quot;</span>, <span class="st">&quot;lme4&quot;</span>, <span class="st">&quot;car&quot;</span>, <span class="st">&quot;gridExtra&quot;</span>, </span>
<span id="cb40-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb40-4" tabindex="-1"></a>               <span class="st">&quot;doParallel&quot;</span>, <span class="st">&quot;ggpubr&quot;</span>, <span class="st">&quot;grid&quot;</span>, <span class="st">&quot;tagcloud&quot;</span>, </span>
<span id="cb40-5"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb40-5" tabindex="-1"></a>               <span class="st">&quot;ggprism&quot;</span>)</span></code></pre></div>
<p>This tutorial also uses a few <a href="https://www.bioconductor.org/">Bioconductor</a> packages. Use <code>BiocManager</code> to install them.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb41-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb41-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb41-2" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb41-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb41-3" tabindex="-1"></a></span>
<span id="cb41-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb41-4" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="fu">c</span>(<span class="st">&quot;DESeq2&quot;</span>,<span class="st">&quot;genomation&quot;</span>,<span class="st">&quot;GenomicFeatures&quot;</span>))</span></code></pre></div>
<p>Load all packages.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb42-1" tabindex="-1"></a>pkgs <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;tidyverse&quot;</span>, <span class="st">&quot;plyr&quot;</span>, <span class="st">&quot;Rfast&quot;</span>, <span class="st">&quot;tryCatchLog&quot;</span>, </span>
<span id="cb42-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb42-2" tabindex="-1"></a>          <span class="st">&quot;lmerTest&quot;</span>, <span class="st">&quot;lme4&quot;</span>, <span class="st">&quot;car&quot;</span>, <span class="st">&quot;gridExtra&quot;</span>, </span>
<span id="cb42-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb42-3" tabindex="-1"></a>          <span class="st">&quot;doParallel&quot;</span>, <span class="st">&quot;ggpubr&quot;</span>, <span class="st">&quot;grid&quot;</span>, <span class="st">&quot;tagcloud&quot;</span>, </span>
<span id="cb42-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb42-4" tabindex="-1"></a>          <span class="st">&quot;ggprism&quot;</span>,<span class="st">&quot;DESeq2&quot;</span>,<span class="st">&quot;genomation&quot;</span>,<span class="st">&quot;GenomicFeatures&quot;</span>)</span>
<span id="cb42-5"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb42-5" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">lapply</span>(pkgs, <span class="cf">function</span>(x) <span class="fu">library</span>(x, <span class="at">character.only=</span><span class="cn">TRUE</span>)))</span></code></pre></div>
</div>
<div id="custom-functions" class="section level3 hasAnchor" number="3.2.3">
<h3><span class="header-section-number">3.2.3</span> Custom functions<a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#custom-functions" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For efficiency, most of the code detailed throughout this tutorial has been compiled as custom functions, which are included in the <a href="https://github.com/sbresnahan/AST-tutorial">project repository</a> as <a href="AST-tutorial/AST_functions.R">AST-tutorial/AST_functions.R</a>. These can be loaded into R during setup:</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb43-1" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;AST-tutorial/AST_functions.R&quot;</span>)</span></code></pre></div>
<p>In this tutorial, each function will first be written explicitly to document how it works before demonstrating its use.</p>
</div>
<div id="sample-metadata" class="section level3 hasAnchor" number="3.2.4">
<h3><span class="header-section-number">3.2.4</span> Sample metadata<a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#sample-metadata" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Many of the custom functions described above reference the sample metadata, which must follow this specific format:</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
sample.id
</th>
<th style="text-align:left;">
parent
</th>
<th style="text-align:left;">
block
</th>
<th style="text-align:left;">
phenotype
</th>
<th style="text-align:left;">
individual
</th>
<th style="text-align:left;">
lineage
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
PARENT_FILE
</td>
<td style="text-align:left;">
M or F
</td>
<td style="text-align:left;">
integer
</td>
<td style="text-align:left;">
character
</td>
<td style="text-align:left;">
FILE
</td>
<td style="text-align:left;">
A or B
</td>
</tr>
</tbody>
</table>
<p>Note: for <code>sample.id</code>, <code>PARENT_FILE</code> corresponds to the file name suffix assigned to the SNP-level read count files generated in Section 2, with <code>PARENT</code> being the respective F1 genome to which the F2 mRNA-seq read <code>FILE</code> was aligned.</p>
<p>Sample metadata for the sequencing data described in <a href="https://doi.org/10.1101/2024.06.09.598129">Bresnahan <em>et al.</em>, 2024, <em>bioRxiv</em></a> used in this tutorial has been included in the <a href="https://github.com/sbresnahan/AST-tutorial">project repository</a> as <a href="AST-tutorial/metadata.csv">AST-tutorial/metadata.csv</a>:</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb44-1" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;AST-tutorial/metadata.csv&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="generate-sample-x-snpgene-count-matrix" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Generate sample x SNP:gene count matrix<a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#generate-sample-x-snpgene-count-matrix" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="filter-snps-by-transcript" class="section level3 hasAnchor" number="3.3.1">
<h3><span class="header-section-number">3.3.1</span> Filter SNPs by transcript<a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#filter-snps-by-transcript" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The honey bee gene annotations saved as <code>$AST-tutorial/{DIR_INDEX}/Amel_HAv3.1_genes.gff3</code> and the F1 SNPs intersecting honey bee genes saved as <code>$AST-tutorial/{DIR_ANALYSIS}/SNPs_for_Analysis_Sorted.bed</code> generated in Section 2 are imported. SNPs are assigned to the longest transcript for each gene, and SNPs within miRNA, tRNA, and repetitive pseudogenes and are discarded. A list of these genes has been included in the <a href="https://github.com/sbresnahan/AST-tutorial">project repository</a> as <a href="AST-tutorial/genelist_filter.csv">AST-tutorial/genelist_filter.csv</a>.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb45-1" tabindex="-1"></a><span class="co"># Load in filter list</span></span>
<span id="cb45-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb45-2" tabindex="-1"></a>filterlist <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;AST-tutorial/genelist_filter.csv&quot;</span>,<span class="at">header=</span>F)[,<span class="fu">c</span>(<span class="dv">1</span>)]</span>
<span id="cb45-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb45-3" tabindex="-1"></a></span>
<span id="cb45-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb45-4" tabindex="-1"></a><span class="co"># Make txdb from the honey bee gene annotations</span></span>
<span id="cb45-5"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb45-5" tabindex="-1"></a>txdb <span class="ot">&lt;-</span> <span class="fu">makeTxDbFromGFF</span>(<span class="st">&quot;AST-tutorial/INDEX/Amel_HAv3.1.gff&quot;</span>)</span>
<span id="cb45-6"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb45-6" tabindex="-1"></a>transcripts <span class="ot">&lt;-</span> <span class="fu">transcripts</span>(txdb)</span></code></pre></div>
<div id="function-filter_snps" class="section level5 unnumbered hasAnchor">
<h5>Function: <code>filter_SNPs</code><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#function-filter_snps" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb46-1" tabindex="-1"></a>filter_SNPs <span class="ot">&lt;-</span> <span class="cf">function</span>(SNP_gene_bed,exons,filterlist){</span>
<span id="cb46-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb46-2" tabindex="-1"></a>  <span class="do">## SNP_gene_bed = bed file of SNPs intersecting genes</span></span>
<span id="cb46-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb46-3" tabindex="-1"></a>  <span class="do">## exons = GRanges object containing exon regions</span></span>
<span id="cb46-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb46-4" tabindex="-1"></a>  <span class="do">## filterlist = list of gene IDs to filter</span></span>
<span id="cb46-5"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb46-5" tabindex="-1"></a></span>
<span id="cb46-6"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb46-6" tabindex="-1"></a>  <span class="co"># Load in the SNPs BED file</span></span>
<span id="cb46-7"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb46-7" tabindex="-1"></a>  SNPs <span class="ot">&lt;-</span> <span class="fu">readBed</span>(SNP_gene_bed)</span>
<span id="cb46-8"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb46-8" tabindex="-1"></a>  <span class="co"># Get SNPs that overlap with transcripts</span></span>
<span id="cb46-9"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb46-9" tabindex="-1"></a>  SNPs.exons <span class="ot">&lt;-</span> <span class="fu">findOverlaps</span>(SNPs,exons)</span>
<span id="cb46-10"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb46-10" tabindex="-1"></a>  SNPs <span class="ot">&lt;-</span> SNPs[<span class="fu">queryHits</span>(SNPs.exons)]</span>
<span id="cb46-11"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb46-11" tabindex="-1"></a>  SNPs <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(SNPs)[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">7</span>)]</span>
<span id="cb46-12"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb46-12" tabindex="-1"></a>  <span class="fu">names</span>(SNPs) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;chr&quot;</span>,<span class="st">&quot;pos&quot;</span>,<span class="st">&quot;SNP_gene&quot;</span>)</span>
<span id="cb46-13"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb46-13" tabindex="-1"></a>  <span class="co"># Create SNP and geneID columns by splitting SNP_gene on &quot;:&quot;</span></span>
<span id="cb46-14"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb46-14" tabindex="-1"></a>  SNPs<span class="sc">$</span>SNP <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">map</span>(<span class="fu">strsplit</span>(SNPs<span class="sc">$</span>SNP_gene,<span class="at">split =</span> <span class="st">&quot;:&quot;</span>), <span class="dv">1</span>))</span>
<span id="cb46-15"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb46-15" tabindex="-1"></a>  SNPs<span class="sc">$</span>geneID <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">map</span>(<span class="fu">strsplit</span>(SNPs<span class="sc">$</span>SNP_gene,<span class="at">split =</span> <span class="st">&quot;:&quot;</span>), <span class="dv">2</span>))</span>
<span id="cb46-16"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb46-16" tabindex="-1"></a>  <span class="co"># Filter SNPs in genes within filterlist</span></span>
<span id="cb46-17"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb46-17" tabindex="-1"></a>  SNPs <span class="ot">&lt;-</span> SNPs[<span class="sc">!</span>SNPs<span class="sc">$</span>geneID<span class="sc">%in%</span>filterlist,]</span>
<span id="cb46-18"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb46-18" tabindex="-1"></a>  <span class="co"># Delete any duplicate rows and clean up</span></span>
<span id="cb46-19"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb46-19" tabindex="-1"></a>  SNPs <span class="ot">&lt;-</span> SNPs[<span class="sc">!</span><span class="fu">duplicated</span>(SNPs),]</span>
<span id="cb46-20"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb46-20" tabindex="-1"></a>  <span class="fu">rm</span>(SNPs.exons)</span>
<span id="cb46-21"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb46-21" tabindex="-1"></a>  </span>
<span id="cb46-22"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb46-22" tabindex="-1"></a>  <span class="co"># Output</span></span>
<span id="cb46-23"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb46-23" tabindex="-1"></a>  <span class="fu">return</span>(SNPs)</span>
<span id="cb46-24"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb46-24" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb47-1" tabindex="-1"></a><span class="co"># Execute</span></span>
<span id="cb47-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb47-2" tabindex="-1"></a>SNPs <span class="ot">&lt;-</span> <span class="fu">filter_SNPs</span>(<span class="st">&quot;AST-tutorial/ANALYSIS_SETS/SNPs_for_Analysis_Sorted.bed&quot;</span>,</span>
<span id="cb47-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb47-3" tabindex="-1"></a>                     transcripts,filterlist)</span>
<span id="cb47-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb47-4" tabindex="-1"></a></span>
<span id="cb47-5"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb47-5" tabindex="-1"></a><span class="co"># Save output</span></span>
<span id="cb47-6"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb47-6" tabindex="-1"></a><span class="fu">write.csv</span>(SNPs,<span class="st">&quot;AST-tutorial/ANALYSIS/INTERMEDIATES/SNPs.csv&quot;</span>,<span class="at">row.names=</span>F)</span></code></pre></div>
<p>Example rows:</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
pos
</th>
<th style="text-align:left;">
SNP_gene
</th>
<th style="text-align:left;">
SNP
</th>
<th style="text-align:left;">
geneID
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
NC_037638.1
</td>
<td style="text-align:right;">
54752
</td>
<td style="text-align:left;">
snp_99616:LOC107964061
</td>
<td style="text-align:left;">
snp_99616
</td>
<td style="text-align:left;">
LOC107964061
</td>
</tr>
<tr>
<td style="text-align:left;">
NC_037638.1
</td>
<td style="text-align:right;">
91397
</td>
<td style="text-align:left;">
snp_99617:LOC113219112
</td>
<td style="text-align:left;">
snp_99617
</td>
<td style="text-align:left;">
LOC113219112
</td>
</tr>
<tr>
<td style="text-align:left;">
NC_037638.1
</td>
<td style="text-align:right;">
147183
</td>
<td style="text-align:left;">
snp_99618:LOC726544
</td>
<td style="text-align:left;">
snp_99618
</td>
<td style="text-align:left;">
LOC726544
</td>
</tr>
<tr>
<td style="text-align:left;">
NC_037638.1
</td>
<td style="text-align:right;">
148247
</td>
<td style="text-align:left;">
snp_99619:LOC726544
</td>
<td style="text-align:left;">
snp_99619
</td>
<td style="text-align:left;">
LOC726544
</td>
</tr>
<tr>
<td style="text-align:left;">
NC_037638.1
</td>
<td style="text-align:right;">
148360
</td>
<td style="text-align:left;">
snp_99620:LOC726544
</td>
<td style="text-align:left;">
snp_99620
</td>
<td style="text-align:left;">
LOC726544
</td>
</tr>
<tr>
<td style="text-align:left;">
NC_037638.1
</td>
<td style="text-align:right;">
148370
</td>
<td style="text-align:left;">
snp_99621:LOC726544
</td>
<td style="text-align:left;">
snp_99621
</td>
<td style="text-align:left;">
LOC726544
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="merge-count-files-to-matrix" class="section level3 hasAnchor" number="3.3.2">
<h3><span class="header-section-number">3.3.2</span> Merge count files to matrix<a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#merge-count-files-to-matrix" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The F2 read counts at F1 SNPs generated in Section 2 are imported. In cases where the distance between SNPs is shorter than the average read length (and thus the read counts are exactly the same for both SNPs), one SNP is chosen at random.</p>
<div id="function-make_ase_counts_matrix" class="section level5 unnumbered hasAnchor">
<h5>Function: <code>make_ASE_counts_matrix</code><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#function-make_ase_counts_matrix" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-1" tabindex="-1"></a>make_ASE_counts_matrix <span class="ot">&lt;-</span> <span class="cf">function</span>(DIR_COUNTS,cores,SNPs){</span>
<span id="cb48-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-2" tabindex="-1"></a>  <span class="do">## DIR_COUNTS = name of subdirectory containing count files</span></span>
<span id="cb48-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-3" tabindex="-1"></a>  <span class="do">## cores = number of threads for parallel processing</span></span>
<span id="cb48-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-4" tabindex="-1"></a>  <span class="do">## SNPs = dataframe generated by filter_SNPs2</span></span>
<span id="cb48-5"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-5" tabindex="-1"></a>  </span>
<span id="cb48-6"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-6" tabindex="-1"></a>  <span class="co"># Read in data</span></span>
<span id="cb48-7"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-7" tabindex="-1"></a>  <span class="do">## List files ending in .txt in DIR_COUNTS</span></span>
<span id="cb48-8"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-8" tabindex="-1"></a>  SNP_counts <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">matrix</span>(<span class="at">ncol=</span><span class="dv">0</span>,<span class="at">nrow=</span><span class="fu">length</span>(SNPs<span class="sc">$</span>SNP_gene)))</span>
<span id="cb48-9"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-9" tabindex="-1"></a>  SNP_counts<span class="sc">$</span>SNP_gene <span class="ot">&lt;-</span> SNPs<span class="sc">$</span>SNP_gene</span>
<span id="cb48-10"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-10" tabindex="-1"></a>  files.counts <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path=</span>DIR_COUNTS, <span class="at">pattern=</span><span class="st">&quot;*.txt&quot;</span>, </span>
<span id="cb48-11"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-11" tabindex="-1"></a>                             <span class="at">full.names=</span><span class="cn">TRUE</span>, <span class="at">recursive=</span><span class="cn">FALSE</span>)</span>
<span id="cb48-12"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-12" tabindex="-1"></a>  <span class="do">## Left join count files by SNP_ID:geneID</span></span>
<span id="cb48-13"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-13" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(files.counts)){</span>
<span id="cb48-14"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-14" tabindex="-1"></a>    <span class="fu">print</span>(i)</span>
<span id="cb48-15"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-15" tabindex="-1"></a>    tmp.name <span class="ot">&lt;-</span>  <span class="fu">strsplit</span>(<span class="fu">strsplit</span>(files.counts[[i]], </span>
<span id="cb48-16"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-16" tabindex="-1"></a>                                   <span class="at">split =</span> <span class="st">&quot;/&quot;</span>)[[<span class="dv">1</span>]][<span class="fu">length</span>(<span class="fu">strsplit</span>(files.counts[[i]], </span>
<span id="cb48-17"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-17" tabindex="-1"></a>                                   <span class="at">split =</span> <span class="st">&quot;/&quot;</span>)[[<span class="dv">1</span>]])],</span>
<span id="cb48-18"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-18" tabindex="-1"></a>                          <span class="at">split =</span> <span class="st">&quot;[.]&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>]</span>
<span id="cb48-19"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-19" tabindex="-1"></a>    <span class="cf">if</span>(tmp.name<span class="sc">%in%</span>metadata<span class="sc">$</span>sample.id){</span>
<span id="cb48-20"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-20" tabindex="-1"></a>      tmp <span class="ot">&lt;-</span> <span class="fu">read.table</span>(files.counts[[i]],<span class="at">header=</span>F)[,<span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">7</span>)]</span>
<span id="cb48-21"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-21" tabindex="-1"></a>      <span class="fu">names</span>(tmp) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;SNP_gene&quot;</span>,tmp.name)</span>
<span id="cb48-22"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-22" tabindex="-1"></a>      SNP_counts <span class="ot">&lt;-</span> SNP_counts <span class="sc">%&gt;%</span> </span>
<span id="cb48-23"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-23" tabindex="-1"></a>        <span class="fu">left_join</span>(tmp, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&#39;SNP_gene&#39;</span> <span class="ot">=</span> <span class="st">&#39;SNP_gene&#39;</span>)) </span>
<span id="cb48-24"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-24" tabindex="-1"></a>    }</span>
<span id="cb48-25"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-25" tabindex="-1"></a>  }</span>
<span id="cb48-26"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-26" tabindex="-1"></a>  <span class="fu">rm</span>(tmp,files.counts,tmp.name)</span>
<span id="cb48-27"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-27" tabindex="-1"></a>  </span>
<span id="cb48-28"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-28" tabindex="-1"></a>  <span class="co"># Clean up dataframe</span></span>
<span id="cb48-29"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-29" tabindex="-1"></a>  <span class="fu">row.names</span>(SNP_counts) <span class="ot">&lt;-</span> SNP_counts<span class="sc">$</span>SNP_gene</span>
<span id="cb48-30"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-30" tabindex="-1"></a>  SNP_counts<span class="sc">$</span>SNP_gene <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb48-31"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-31" tabindex="-1"></a>  SNP_counts[<span class="fu">is.na</span>(SNP_counts)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb48-32"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-32" tabindex="-1"></a>  SNP_counts<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">map</span>(<span class="fu">strsplit</span>(<span class="fu">row.names</span>(SNP_counts), <span class="at">split =</span> <span class="st">&quot;:&quot;</span>), <span class="dv">2</span>))</span>
<span id="cb48-33"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-33" tabindex="-1"></a>  </span>
<span id="cb48-34"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-34" tabindex="-1"></a>  <span class="co"># Remove duplicate rows within genes</span></span>
<span id="cb48-35"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-35" tabindex="-1"></a>  genelist <span class="ot">&lt;-</span> <span class="fu">unique</span>(SNP_counts<span class="sc">$</span>gene)</span>
<span id="cb48-36"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-36" tabindex="-1"></a>  delete.rows <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb48-37"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-37" tabindex="-1"></a>  <span class="do">## Use DoParallel to perform loop operation in parallel</span></span>
<span id="cb48-38"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-38" tabindex="-1"></a>  <span class="fu">registerDoParallel</span>(<span class="at">cores=</span>cores)</span>
<span id="cb48-39"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-39" tabindex="-1"></a>  delete.rows <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i=</span><span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(genelist)) <span class="sc">%dopar%</span> {</span>
<span id="cb48-40"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-40" tabindex="-1"></a>    tmp <span class="ot">&lt;-</span> SNP_counts[SNP_counts<span class="sc">$</span>gene<span class="sc">==</span>genelist[i],]</span>
<span id="cb48-41"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-41" tabindex="-1"></a>    d <span class="ot">&lt;-</span> <span class="fu">row.names</span>(tmp[<span class="fu">duplicated</span>(tmp),])</span>
<span id="cb48-42"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-42" tabindex="-1"></a>    delete.rows <span class="ot">&lt;-</span> <span class="fu">return</span>(d)</span>
<span id="cb48-43"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-43" tabindex="-1"></a>  }</span>
<span id="cb48-44"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-44" tabindex="-1"></a>  SNP_counts <span class="ot">&lt;-</span> SNP_counts[<span class="sc">!</span><span class="fu">row.names</span>(SNP_counts)<span class="sc">%in%</span><span class="fu">unlist</span>(delete.rows),]</span>
<span id="cb48-45"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-45" tabindex="-1"></a>  </span>
<span id="cb48-46"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-46" tabindex="-1"></a>  <span class="co"># Output</span></span>
<span id="cb48-47"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-47" tabindex="-1"></a>  <span class="fu">return</span>(SNP_counts)</span>
<span id="cb48-48"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb48-48" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb49-1" tabindex="-1"></a><span class="co"># Execute</span></span>
<span id="cb49-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb49-2" tabindex="-1"></a>SNP_counts <span class="ot">&lt;-</span> <span class="fu">make_ASE_counts_matrix</span>(<span class="st">&quot;AST-tutorial/COUNT&quot;</span>,<span class="dv">4</span>,SNPs)</span>
<span id="cb49-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb49-3" tabindex="-1"></a></span>
<span id="cb49-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb49-4" tabindex="-1"></a><span class="co"># Save output</span></span>
<span id="cb49-5"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb49-5" tabindex="-1"></a><span class="fu">write.csv</span>(SNP_counts,<span class="st">&quot;AST-tutorial/ANALYSIS/INTERMEDIATES/SNP_gene_counts.csv&quot;</span>)</span></code></pre></div>
<p>Example rows:</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
B3D_CMG0017_S9
</th>
<th style="text-align:right;">
B3D_CMG0018_S10
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
snp_99616:LOC107964061
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
7
</td>
</tr>
<tr>
<td style="text-align:left;">
snp_99617:LOC113219112
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
snp_99618:LOC726544
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
11
</td>
</tr>
<tr>
<td style="text-align:left;">
snp_99619:LOC726544
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
7
</td>
</tr>
<tr>
<td style="text-align:left;">
snp_1:LOC726544
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
7
</td>
</tr>
<tr>
<td style="text-align:left;">
snp_99623:LOC726544
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
8
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="prepare-sample-x-snpgene-counts-for-analysis" class="section level2 hasAnchor" number="3.4">
<h2><span class="header-section-number">3.4</span> Prepare sample x SNP:gene counts for analysis<a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#prepare-sample-x-snpgene-counts-for-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="normalize-by-library-size" class="section level3 hasAnchor" number="3.4.1">
<h3><span class="header-section-number">3.4.1</span> Normalize by library size<a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#normalize-by-library-size" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In spite of every effort to standardize the library preparation and sequencing procedures, there are innumerable sources of between-sample variation that cannot be controlled for which will result in variation in sequencing depth between libraries. This is apparent here:</p>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
<p>To adjust for differences in sequencing depth between libraries, library size factors are estimated and the read counts are normalized using the median of ratios normalization (MRN) method from DESeq2.</p>
<p><strong>First</strong>, sum the parent-specific counts by library and estimate library size factors:</p>
<div id="function-calcsizefactors" class="section level5 unnumbered hasAnchor">
<h5>Function: <code>calcSizeFactors</code><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#function-calcsizefactors" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-1" tabindex="-1"></a>calcSizeFactors <span class="ot">&lt;-</span> <span class="cf">function</span>(counts,mdata){</span>
<span id="cb50-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-2" tabindex="-1"></a>  <span class="do">## counts = counts matrix generated by make_ASE_counts_matrix</span></span>
<span id="cb50-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-3" tabindex="-1"></a>  <span class="do">## mdata = dataframe containing sample metadata</span></span>
<span id="cb50-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-4" tabindex="-1"></a>  </span>
<span id="cb50-5"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-5" tabindex="-1"></a>  <span class="co"># Merge parent-specific counts by library</span></span>
<span id="cb50-6"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-6" tabindex="-1"></a>  counts <span class="ot">&lt;-</span> counts[,<span class="fu">names</span>(counts)<span class="sc">%in%</span>mdata<span class="sc">$</span>sample.id]</span>
<span id="cb50-7"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-7" tabindex="-1"></a>  counts_merged <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">matrix</span>(<span class="at">ncol=</span><span class="dv">0</span>,<span class="at">nrow=</span><span class="fu">length</span>(<span class="fu">row.names</span>(counts))))</span>
<span id="cb50-8"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-8" tabindex="-1"></a>  samples <span class="ot">&lt;-</span> <span class="fu">unique</span>(mdata<span class="sc">$</span>individual)</span>
<span id="cb50-9"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-9" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(samples)){</span>
<span id="cb50-10"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-10" tabindex="-1"></a>    tmp <span class="ot">&lt;-</span> counts[,<span class="fu">names</span>(counts)<span class="sc">%in%</span>mdata[mdata<span class="sc">$</span>individual<span class="sc">==</span>samples[i],<span class="st">&quot;sample.id&quot;</span>]]</span>
<span id="cb50-11"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-11" tabindex="-1"></a>    tmp.merged <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(tmp)</span>
<span id="cb50-12"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-12" tabindex="-1"></a>    counts_merged <span class="ot">&lt;-</span> <span class="fu">cbind</span>(counts_merged,tmp.merged)</span>
<span id="cb50-13"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-13" tabindex="-1"></a>  }</span>
<span id="cb50-14"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-14" tabindex="-1"></a>  <span class="fu">names</span>(counts_merged) <span class="ot">&lt;-</span> samples</span>
<span id="cb50-15"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-15" tabindex="-1"></a>  mdata.merged <span class="ot">&lt;-</span> mdata[,<span class="fu">c</span>(<span class="st">&quot;individual&quot;</span>,</span>
<span id="cb50-16"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-16" tabindex="-1"></a>                           <span class="st">&quot;phenotype&quot;</span>,</span>
<span id="cb50-17"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-17" tabindex="-1"></a>                           <span class="st">&quot;lineage&quot;</span>)]</span>
<span id="cb50-18"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-18" tabindex="-1"></a>  mdata.merged <span class="ot">&lt;-</span> mdata.merged[<span class="sc">!</span><span class="fu">duplicated</span>(mdata.merged),]</span>
<span id="cb50-19"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-19" tabindex="-1"></a>  </span>
<span id="cb50-20"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-20" tabindex="-1"></a>  <span class="co"># Estimate library size factors using the median of ratios normalization method from DESeq2</span></span>
<span id="cb50-21"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-21" tabindex="-1"></a>  dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> counts_merged, <span class="at">colData =</span> mdata.merged, </span>
<span id="cb50-22"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-22" tabindex="-1"></a>                                <span class="at">design =</span> <span class="sc">~</span> lineage <span class="sc">+</span> phenotype)</span>
<span id="cb50-23"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-23" tabindex="-1"></a>  dds <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(dds)</span>
<span id="cb50-24"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-24" tabindex="-1"></a>  sizeFactors <span class="ot">&lt;-</span> <span class="fu">sizeFactors</span>(dds)</span>
<span id="cb50-25"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-25" tabindex="-1"></a>  sF2meta <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">individual=</span><span class="fu">names</span>(sizeFactors))</span>
<span id="cb50-26"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-26" tabindex="-1"></a>  sF2meta<span class="sc">$</span>sF <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(sizeFactors)</span>
<span id="cb50-27"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-27" tabindex="-1"></a>  sF2meta <span class="ot">&lt;-</span> <span class="fu">left_join</span>(sF2meta,mdata[,<span class="fu">c</span>(<span class="st">&quot;individual&quot;</span>,</span>
<span id="cb50-28"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-28" tabindex="-1"></a>                                        <span class="st">&quot;sample.id&quot;</span>)],</span>
<span id="cb50-29"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-29" tabindex="-1"></a>                       <span class="at">multiple =</span> <span class="st">&quot;all&quot;</span>)</span>
<span id="cb50-30"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-30" tabindex="-1"></a>  sF2meta <span class="ot">&lt;-</span> sF2meta[,<span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">2</span>)]</span>
<span id="cb50-31"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-31" tabindex="-1"></a>  sF2meta <span class="ot">&lt;-</span> sF2meta[<span class="fu">match</span>(<span class="fu">names</span>(counts), sF2meta<span class="sc">$</span>sample.id),]</span>
<span id="cb50-32"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-32" tabindex="-1"></a>  sFs <span class="ot">&lt;-</span> sF2meta<span class="sc">$</span>sF</span>
<span id="cb50-33"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-33" tabindex="-1"></a>  <span class="fu">names</span>(sFs) <span class="ot">&lt;-</span> sF2meta<span class="sc">$</span>sample.id</span>
<span id="cb50-34"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-34" tabindex="-1"></a>  </span>
<span id="cb50-35"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-35" tabindex="-1"></a>  <span class="co"># Output</span></span>
<span id="cb50-36"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-36" tabindex="-1"></a>  <span class="fu">return</span>(sFs)</span>
<span id="cb50-37"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb50-37" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb51-1" tabindex="-1"></a><span class="co"># Execute</span></span>
<span id="cb51-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb51-2" tabindex="-1"></a>size_factors <span class="ot">&lt;-</span> <span class="fu">calcSizeFactors</span>(SNP_counts,metadata)</span></code></pre></div>
<p><strong>Second</strong>, perform the MRN method on the parent-specific counts using the library size factors calculated above.</p>
</div>
<div id="function-normalizeasreadcounts" class="section level5 unnumbered hasAnchor">
<h5>Function: <code>normalizeASReadCounts</code><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#function-normalizeasreadcounts" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb52-1" tabindex="-1"></a>normalizeASReadCounts <span class="ot">&lt;-</span> <span class="cf">function</span>(counts,size_factors){</span>
<span id="cb52-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb52-2" tabindex="-1"></a>  <span class="do">## counts = counts matrix generated by make_ASE_counts_matrix</span></span>
<span id="cb52-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb52-3" tabindex="-1"></a>  <span class="do">## size_factors = array from calcSizeFactors</span></span>
<span id="cb52-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb52-4" tabindex="-1"></a>  </span>
<span id="cb52-5"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb52-5" tabindex="-1"></a>  <span class="co"># Normalize counts for each library by parent</span></span>
<span id="cb52-6"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb52-6" tabindex="-1"></a>  mdata <span class="ot">&lt;-</span> metadata[metadata<span class="sc">$</span>sample.id<span class="sc">%in%</span><span class="fu">names</span>(counts),]</span>
<span id="cb52-7"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb52-7" tabindex="-1"></a>  counts <span class="ot">&lt;-</span> counts[,<span class="fu">names</span>(counts)<span class="sc">%in%</span>mdata<span class="sc">$</span>sample.id]</span>
<span id="cb52-8"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb52-8" tabindex="-1"></a>  as.dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> counts, <span class="at">colData =</span> mdata, </span>
<span id="cb52-9"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb52-9" tabindex="-1"></a>                                   <span class="at">design =</span> <span class="sc">~</span> lineage<span class="sc">+</span>phenotype)</span>
<span id="cb52-10"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb52-10" tabindex="-1"></a>  <span class="fu">sizeFactors</span>(as.dds) <span class="ot">=</span> size_factors</span>
<span id="cb52-11"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb52-11" tabindex="-1"></a>  counts_normalized <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">counts</span>(as.dds, <span class="at">normalized=</span><span class="cn">TRUE</span>))</span>
<span id="cb52-12"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb52-12" tabindex="-1"></a>  </span>
<span id="cb52-13"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb52-13" tabindex="-1"></a>  <span class="co"># Output</span></span>
<span id="cb52-14"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb52-14" tabindex="-1"></a>  <span class="fu">return</span>(counts_normalized)</span>
<span id="cb52-15"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb52-15" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb53-1" tabindex="-1"></a><span class="co"># Execute</span></span>
<span id="cb53-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb53-2" tabindex="-1"></a>SNP_counts_normalized <span class="ot">&lt;-</span> <span class="fu">normalizeASReadCounts</span>(SNP_counts,size_factors)</span>
<span id="cb53-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb53-3" tabindex="-1"></a></span>
<span id="cb53-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb53-4" tabindex="-1"></a><span class="co"># Save output</span></span>
<span id="cb53-5"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb53-5" tabindex="-1"></a><span class="fu">write.csv</span>(SNP_counts_normalized,<span class="st">&quot;AST-tutorial/ANALYSIS/INTERMEDIATES/SNP_gene_counts_normalized.csv&quot;</span>)</span></code></pre></div>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-22-1.png" width="672" /></p>
</div>
</div>
<div id="split-by-phenotype-and-filter-low-count-snps" class="section level3 hasAnchor" number="3.4.2">
<h3><span class="header-section-number">3.4.2</span> Split by phenotype and filter low count SNPs<a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#split-by-phenotype-and-filter-low-count-snps" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Further filtering steps are performed <em>after</em> splitting the count matrix by phenotype. This is important as the set of transcribed genes is expected to vary between phenotypes. For the remainder of the tutorial, these phenotypes will be referred to in the code as “WL” and “QL” to match the sample metadata.</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb54-1" tabindex="-1"></a>WL.IDs <span class="ot">&lt;-</span> metadata[metadata<span class="sc">$</span>phenotype<span class="sc">==</span><span class="st">&quot;WL&quot;</span>,<span class="st">&quot;sample.id&quot;</span>]</span>
<span id="cb54-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb54-2" tabindex="-1"></a>WL_counts_normalized <span class="ot">&lt;-</span> SNP_counts_normalized[,<span class="fu">names</span>(SNP_counts_normalized)<span class="sc">%in%</span>WL.IDs]</span>
<span id="cb54-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb54-3" tabindex="-1"></a>WL_counts <span class="ot">&lt;-</span> SNP_counts[<span class="fu">row.names</span>(SNP_counts)<span class="sc">%in%</span><span class="fu">row.names</span>(WL_counts_normalized),</span>
<span id="cb54-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb54-4" tabindex="-1"></a>                                  <span class="fu">names</span>(SNP_counts)<span class="sc">%in%</span>WL.IDs]</span>
<span id="cb54-5"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb54-5" tabindex="-1"></a></span>
<span id="cb54-6"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb54-6" tabindex="-1"></a>QL.IDs <span class="ot">&lt;-</span> metadata[metadata<span class="sc">$</span>phenotype<span class="sc">==</span><span class="st">&quot;QL&quot;</span>,<span class="st">&quot;sample.id&quot;</span>]</span>
<span id="cb54-7"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb54-7" tabindex="-1"></a>QL_counts_normalized <span class="ot">&lt;-</span> SNP_counts_normalized[,<span class="fu">names</span>(SNP_counts_normalized)<span class="sc">%in%</span>QL.IDs]</span>
<span id="cb54-8"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb54-8" tabindex="-1"></a>QL_counts <span class="ot">&lt;-</span> SNP_counts[<span class="fu">row.names</span>(SNP_counts)<span class="sc">%in%</span><span class="fu">row.names</span>(QL_counts_normalized),</span>
<span id="cb54-9"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb54-9" tabindex="-1"></a>                                <span class="fu">names</span>(SNP_counts)<span class="sc">%in%</span>QL.IDs]</span></code></pre></div>
<p>SNPs with <span class="math inline">\(n &lt; 1\)</span> count in any Lineage are discarded. Finally, genes with <span class="math inline">\(n &lt; 2\)</span> SNPs are discarded, as this method requires at least two independent observations per gene for statistical analysis.</p>
<p><em>Note: here is where the utility of functions really becomes apparent. For the remainder of the tutorial, the same functions will be applied to the “WL” and “QL” count matrices.</em></p>
<div id="function-filter_counts" class="section level5 unnumbered hasAnchor">
<h5>Function: <code>filter_counts</code><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#function-filter_counts" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-1" tabindex="-1"></a>filter_counts <span class="ot">&lt;-</span> <span class="cf">function</span>(counts,lcf){</span>
<span id="cb55-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-2" tabindex="-1"></a>  <span class="co"># counts = phenotype-specific counts matrix</span></span>
<span id="cb55-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-3" tabindex="-1"></a>  <span class="co"># lcf = low count filter threshold (integer)</span></span>
<span id="cb55-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-4" tabindex="-1"></a>  </span>
<span id="cb55-5"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-5" tabindex="-1"></a>  <span class="co"># Remove rows with &lt; lcf counts counts by Lineage</span></span>
<span id="cb55-6"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-6" tabindex="-1"></a>  LA <span class="ot">&lt;-</span> metadata[metadata<span class="sc">$</span>lineage<span class="sc">==</span><span class="st">&quot;A&quot;</span>,<span class="st">&quot;sample.id&quot;</span>]</span>
<span id="cb55-7"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-7" tabindex="-1"></a>  LB <span class="ot">&lt;-</span> metadata[metadata<span class="sc">$</span>lineage<span class="sc">==</span><span class="st">&quot;B&quot;</span>,<span class="st">&quot;sample.id&quot;</span>]</span>
<span id="cb55-8"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-8" tabindex="-1"></a>  counts <span class="ot">&lt;-</span> counts[<span class="fu">rowSums</span>(counts[,<span class="fu">names</span>(counts)<span class="sc">%in%</span>LA])<span class="sc">&gt;</span>lcf,]</span>
<span id="cb55-9"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-9" tabindex="-1"></a>  counts <span class="ot">&lt;-</span> counts[<span class="fu">rowSums</span>(counts[,<span class="fu">names</span>(counts)<span class="sc">%in%</span>LB])<span class="sc">&gt;</span>lcf,]</span>
<span id="cb55-10"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-10" tabindex="-1"></a>  </span>
<span id="cb55-11"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-11" tabindex="-1"></a>  <span class="co"># Flag rows with greater than 10000 counts</span></span>
<span id="cb55-12"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-12" tabindex="-1"></a>  <span class="do">## Different functions are used for SK tests with &lt; and &gt; 10000 counts for computational efficiency</span></span>
<span id="cb55-13"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-13" tabindex="-1"></a>  counts<span class="sc">$</span>SUM <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(counts)</span>
<span id="cb55-14"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-14" tabindex="-1"></a>  counts<span class="sc">$</span>SKrow <span class="ot">&lt;-</span> F</span>
<span id="cb55-15"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-15" tabindex="-1"></a>  counts[counts<span class="sc">$</span>SUM<span class="sc">&lt;</span><span class="dv">10000</span>,<span class="st">&quot;SKrow&quot;</span>] <span class="ot">&lt;-</span> T</span>
<span id="cb55-16"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-16" tabindex="-1"></a>  counts<span class="sc">$</span>SUM <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb55-17"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-17" tabindex="-1"></a>  </span>
<span id="cb55-18"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-18" tabindex="-1"></a>  <span class="co"># Remove genes with &lt; 2 SNPs</span></span>
<span id="cb55-19"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-19" tabindex="-1"></a>  counts<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">map</span>(<span class="fu">strsplit</span>(<span class="fu">row.names</span>(counts), <span class="at">split =</span> <span class="st">&quot;:&quot;</span>), <span class="dv">2</span>))</span>
<span id="cb55-20"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-20" tabindex="-1"></a>  genelist <span class="ot">&lt;-</span> <span class="fu">unique</span>(counts<span class="sc">$</span>gene)</span>
<span id="cb55-21"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-21" tabindex="-1"></a>  delete.rows <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb55-22"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-22" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(genelist)){</span>
<span id="cb55-23"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-23" tabindex="-1"></a>    tmp <span class="ot">&lt;-</span> counts[counts<span class="sc">$</span>gene<span class="sc">==</span>genelist[i],]</span>
<span id="cb55-24"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-24" tabindex="-1"></a>    tmp <span class="ot">&lt;-</span> tmp[<span class="sc">!</span><span class="fu">duplicated</span>(tmp),]</span>
<span id="cb55-25"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-25" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(<span class="fu">row.names</span>(tmp))<span class="sc">&lt;</span><span class="dv">2</span>){</span>
<span id="cb55-26"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-26" tabindex="-1"></a>      delete.rows <span class="ot">&lt;-</span> <span class="fu">append</span>(delete.rows,genelist[i])</span>
<span id="cb55-27"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-27" tabindex="-1"></a>    }</span>
<span id="cb55-28"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-28" tabindex="-1"></a>  }</span>
<span id="cb55-29"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-29" tabindex="-1"></a>  counts <span class="ot">&lt;-</span> counts[<span class="sc">!</span>counts<span class="sc">$</span>gene<span class="sc">%in%</span><span class="fu">unlist</span>(delete.rows),]</span>
<span id="cb55-30"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-30" tabindex="-1"></a>  counts<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb55-31"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-31" tabindex="-1"></a>  </span>
<span id="cb55-32"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-32" tabindex="-1"></a>  <span class="co"># Return filtered counts</span></span>
<span id="cb55-33"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-33" tabindex="-1"></a>  <span class="fu">return</span>(counts)</span>
<span id="cb55-34"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb55-34" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb56-1" tabindex="-1"></a><span class="co"># Execute</span></span>
<span id="cb56-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb56-2" tabindex="-1"></a>WL_counts_normalized <span class="ot">&lt;-</span> <span class="fu">filter_counts</span>(WL_counts_normalized,<span class="dv">0</span>)</span>
<span id="cb56-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb56-3" tabindex="-1"></a>WL_counts <span class="ot">&lt;-</span> WL_counts[<span class="fu">row.names</span>(WL_counts)<span class="sc">%in%</span><span class="fu">row.names</span>(WL_counts_normalized),]</span>
<span id="cb56-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb56-4" tabindex="-1"></a>QL_counts_normalized <span class="ot">&lt;-</span> <span class="fu">filter_counts</span>(QL_counts_normalized,<span class="dv">0</span>)</span>
<span id="cb56-5"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb56-5" tabindex="-1"></a>QL_counts <span class="ot">&lt;-</span> QL_counts[<span class="fu">row.names</span>(QL_counts)<span class="sc">%in%</span><span class="fu">row.names</span>(QL_counts_normalized),]</span>
<span id="cb56-6"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb56-6" tabindex="-1"></a></span>
<span id="cb56-7"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb56-7" tabindex="-1"></a><span class="co"># Save output</span></span>
<span id="cb56-8"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb56-8" tabindex="-1"></a><span class="fu">write.csv</span>(WL_counts_normalized,<span class="st">&quot;AST-tutorial/ANALYSIS/INTERMEDIATES/G53xY39_WL_counts_normalized.csv&quot;</span>)</span>
<span id="cb56-9"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb56-9" tabindex="-1"></a><span class="fu">write.csv</span>(WL_counts,<span class="st">&quot;AST-tutorial/ANALYSIS/INTERMEDIATES/G53xY39_WL_counts.csv&quot;</span>)</span>
<span id="cb56-10"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb56-10" tabindex="-1"></a><span class="fu">write.csv</span>(QL_counts_normalized,<span class="st">&quot;AST-tutorial/ANALYSIS/INTERMEDIATES/G53xY39_QL_counts_normalized.csv&quot;</span>)</span>
<span id="cb56-11"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb56-11" tabindex="-1"></a><span class="fu">write.csv</span>(QL_counts,<span class="st">&quot;AST-tutorial/ANALYSIS/INTERMEDIATES/G53xY39_QL_counts.csv&quot;</span>)</span></code></pre></div>
</div>
</div>
</div>
<div id="conduct-statistical-tests" class="section level2 hasAnchor" number="3.5">
<h2><span class="header-section-number">3.5</span> Conduct statistical tests<a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#conduct-statistical-tests" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The data are now ready for statistical analysis of allele-specific transcription. Storer-Kim, GLIMMIX, and threshold tests will be performed for each phenotype, separately, and then a Chi-squared test will be performed to compare the distribution of genes showing parent- and lineage-biased transcription between phenotypes.</p>
<div id="snp-level-storer-kim-sk-binomial-exact-tests" class="section level3 hasAnchor" number="3.5.1">
<h3><span class="header-section-number">3.5.1</span> SNP-level Storer-Kim (SK) binomial exact tests<a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#snp-level-storer-kim-sk-binomial-exact-tests" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For each SNP, a <a href="https://doi.org/10.1080/01621459.1990.10475318">Storer-Kim binomial exact test of two proportions</a> is conducted using the MRN counts to test the hypothesis that the proportion of maternal and paternal read counts are statistically different.</p>
<div id="function-twobinom" class="section level5 unnumbered hasAnchor">
<h5>Function: <code>twobinom</code><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#function-twobinom" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>This test is performed using a modified version of the <code>twobinom</code> function from <a href="https://cran.r-project.org/web/packages/WRS2/index.html">WRS2</a>. For computational efficiency, the <code>outer()</code> command has been replaced with <code>Rfast::Outer()</code>.</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-1" tabindex="-1"></a>twobinom<span class="ot">&lt;-</span><span class="cf">function</span>(r1,n1,r2,n2,<span class="at">alpha=</span>.<span class="dv">05</span>){</span>
<span id="cb57-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-2" tabindex="-1"></a>  <span class="co"># r1 = success in group 1</span></span>
<span id="cb57-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-3" tabindex="-1"></a>  <span class="co"># n1 = total in group 1</span></span>
<span id="cb57-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-4" tabindex="-1"></a>  <span class="co"># r2 = success in group 2</span></span>
<span id="cb57-5"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-5" tabindex="-1"></a>  <span class="co"># r2 = total in group 2</span></span>
<span id="cb57-6"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-6" tabindex="-1"></a>  n1p<span class="ot">&lt;-</span>n1<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb57-7"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-7" tabindex="-1"></a>  n2p<span class="ot">&lt;-</span>n2<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb57-8"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-8" tabindex="-1"></a>  n1m<span class="ot">&lt;-</span>n1<span class="dv">-1</span></span>
<span id="cb57-9"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-9" tabindex="-1"></a>  n2m<span class="ot">&lt;-</span>n2<span class="dv">-1</span></span>
<span id="cb57-10"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-10" tabindex="-1"></a>  q <span class="ot">&lt;-</span> r1<span class="sc">/</span>n1</span>
<span id="cb57-11"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-11" tabindex="-1"></a>  p <span class="ot">&lt;-</span> r2<span class="sc">/</span>n2</span>
<span id="cb57-12"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-12" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">is.na</span>(q)){q <span class="ot">&lt;-</span> <span class="dv">0</span>}</span>
<span id="cb57-13"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-13" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">is.na</span>(p)){p <span class="ot">&lt;-</span> <span class="dv">0</span>}</span>
<span id="cb57-14"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-14" tabindex="-1"></a>  chk<span class="ot">&lt;-</span><span class="fu">abs</span>(q<span class="sc">-</span>p)</span>
<span id="cb57-15"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-15" tabindex="-1"></a>  x<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="dv">0</span><span class="sc">:</span>n1)<span class="sc">/</span>n1</span>
<span id="cb57-16"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-16" tabindex="-1"></a>  y<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="dv">0</span><span class="sc">:</span>n2)<span class="sc">/</span>n2  </span>
<span id="cb57-17"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-17" tabindex="-1"></a>  phat<span class="ot">&lt;-</span>(r1<span class="sc">+</span>r2)<span class="sc">/</span>(n1<span class="sc">+</span>n2)</span>
<span id="cb57-18"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-18" tabindex="-1"></a>  m1<span class="ot">&lt;-</span><span class="fu">t</span>(<span class="fu">Outer</span>(x,y,<span class="st">&quot;-&quot;</span>))</span>
<span id="cb57-19"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-19" tabindex="-1"></a>  m2<span class="ot">&lt;-</span><span class="fu">matrix</span>(<span class="dv">1</span>,n1p,n2p)</span>
<span id="cb57-20"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-20" tabindex="-1"></a>  flag<span class="ot">&lt;-</span>(<span class="fu">abs</span>(m1)<span class="sc">&gt;=</span>chk)</span>
<span id="cb57-21"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-21" tabindex="-1"></a>  m3<span class="ot">&lt;-</span>m2<span class="sc">*</span>flag</span>
<span id="cb57-22"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-22" tabindex="-1"></a>  <span class="fu">rm</span>(m1,m2,flag)</span>
<span id="cb57-23"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-23" tabindex="-1"></a>  xv<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span>n1)</span>
<span id="cb57-24"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-24" tabindex="-1"></a>  yv<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span>n2)</span>
<span id="cb57-25"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-25" tabindex="-1"></a>  xv1<span class="ot">&lt;-</span>n1<span class="sc">-</span>xv<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb57-26"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-26" tabindex="-1"></a>  yv1<span class="ot">&lt;-</span>n2<span class="sc">-</span>yv<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb57-27"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-27" tabindex="-1"></a>  dis1<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="fu">pbeta</span>(phat,xv,xv1))</span>
<span id="cb57-28"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-28" tabindex="-1"></a>  dis2<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="fu">pbeta</span>(phat,yv,yv1))</span>
<span id="cb57-29"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-29" tabindex="-1"></a>  pd1<span class="ot">&lt;-</span><span class="cn">NA</span></span>
<span id="cb57-30"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-30" tabindex="-1"></a>  pd2<span class="ot">&lt;-</span><span class="cn">NA</span></span>
<span id="cb57-31"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-31" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n1){pd1[i]<span class="ot">&lt;-</span>dis1[i]<span class="sc">-</span>dis1[i<span class="sc">+</span><span class="dv">1</span>]}</span>
<span id="cb57-32"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-32" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n2){pd2[i]<span class="ot">&lt;-</span>dis2[i]<span class="sc">-</span>dis2[i<span class="sc">+</span><span class="dv">1</span>]}</span>
<span id="cb57-33"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-33" tabindex="-1"></a>  pd1[n1p]<span class="ot">&lt;-</span>phat<span class="sc">^</span>n1</span>
<span id="cb57-34"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-34" tabindex="-1"></a>  pd2[n2p]<span class="ot">&lt;-</span>phat<span class="sc">^</span>n2</span>
<span id="cb57-35"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-35" tabindex="-1"></a>  m4<span class="ot">&lt;-</span><span class="fu">t</span>(<span class="fu">Outer</span>(pd1,pd2,<span class="st">&quot;*&quot;</span>))</span>
<span id="cb57-36"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-36" tabindex="-1"></a>  test<span class="ot">&lt;-</span><span class="fu">sum</span>(m3<span class="sc">*</span>m4)</span>
<span id="cb57-37"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-37" tabindex="-1"></a>  <span class="fu">rm</span>(m3,m4)</span>
<span id="cb57-38"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-38" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">p.value=</span>test,<span class="at">p1=</span>q,<span class="at">p2=</span>p,<span class="at">est.dif=</span>q<span class="sc">-</span>p)</span>
<span id="cb57-39"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb57-39" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="function-ast.sk" class="section level5 unnumbered hasAnchor">
<h5>Function: <code>AST.SK</code><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#function-ast.sk" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>This is a wrapper function to perform the Storer-Kim test comparing the proportion of maternal to paternal read counts on each SNP in the count matrix. The Storer-Kim test has greater statistical power for low count observations, but is computationally expensive to perform and crashes when the total counts for a SNP are <em>n</em> &gt; 10,000. Therefore, <code>filter_counts</code> flagged these rows, and <code>AST.SK</code> will instead perform a Fisher’s Exact test of two proportions, which is equivalent to a Storer-Kim test for high count observations. The test <em>p</em>-value is reported for each SNP.</p>
<p><em>Note: because this test is computationally expensive, executing <code>AST.SK</code> will take approximately 2hrs per 10,000 SNPs.</em></p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-1" tabindex="-1"></a>AST.SK <span class="ot">&lt;-</span> <span class="cf">function</span>(counts,phenotype,cores){</span>
<span id="cb58-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-2" tabindex="-1"></a>  <span class="co"># counts = phenotype-specific counts matrix</span></span>
<span id="cb58-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-3" tabindex="-1"></a>  <span class="co"># phenotype = corresponding phenotype of counts</span></span>
<span id="cb58-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-4" tabindex="-1"></a>  <span class="co"># cores = # of threads for multithreaded search</span></span>
<span id="cb58-5"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-5" tabindex="-1"></a>  </span>
<span id="cb58-6"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-6" tabindex="-1"></a>  <span class="co"># Split data by pat and mat</span></span>
<span id="cb58-7"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-7" tabindex="-1"></a>  pat.exp <span class="ot">&lt;-</span> counts[,metadata[metadata<span class="sc">$</span>parent<span class="sc">%in%</span><span class="fu">c</span>(<span class="st">&quot;M&quot;</span>)<span class="sc">&amp;</span></span>
<span id="cb58-8"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-8" tabindex="-1"></a>                                metadata<span class="sc">$</span>phenotype<span class="sc">==</span>phenotype,<span class="st">&quot;sample.id&quot;</span>]]</span>
<span id="cb58-9"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-9" tabindex="-1"></a>  mat.exp <span class="ot">&lt;-</span> counts[,metadata[metadata<span class="sc">$</span>parent<span class="sc">%in%</span><span class="fu">c</span>(<span class="st">&quot;F&quot;</span>)<span class="sc">&amp;</span></span>
<span id="cb58-10"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-10" tabindex="-1"></a>                                metadata<span class="sc">$</span>phenotype<span class="sc">==</span>phenotype,<span class="st">&quot;sample.id&quot;</span>]]</span>
<span id="cb58-11"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-11" tabindex="-1"></a>  <span class="co"># Set up for DoParallel</span></span>
<span id="cb58-12"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-12" tabindex="-1"></a>  SKrows<span class="ot">=</span>counts<span class="sc">$</span>SKrow</span>
<span id="cb58-13"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-13" tabindex="-1"></a>  <span class="fu">registerDoParallel</span>(<span class="at">cores=</span>cores)</span>
<span id="cb58-14"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-14" tabindex="-1"></a>  i.len<span class="ot">=</span><span class="fu">length</span>(<span class="fu">row.names</span>(pat.exp))</span>
<span id="cb58-15"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-15" tabindex="-1"></a>  </span>
<span id="cb58-16"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-16" tabindex="-1"></a>  <span class="co"># For each row, conduct an SK test and return the p-value</span></span>
<span id="cb58-17"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-17" tabindex="-1"></a>  return.df <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i=</span><span class="dv">1</span><span class="sc">:</span>i.len, <span class="at">.combine=</span>rbind,</span>
<span id="cb58-18"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-18" tabindex="-1"></a>                       <span class="at">.export=</span><span class="fu">ls</span>(<span class="fu">globalenv</span>()),<span class="at">.packages=</span><span class="st">&quot;Rfast&quot;</span>) <span class="sc">%dopar%</span> {</span>
<span id="cb58-19"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-19" tabindex="-1"></a>                         SNP_gene<span class="ot">=</span><span class="fu">row.names</span>(pat.exp[i,])</span>
<span id="cb58-20"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-20" tabindex="-1"></a>                         p1.s<span class="ot">=</span><span class="fu">sum</span>(pat.exp[i,])</span>
<span id="cb58-21"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-21" tabindex="-1"></a>                         p2.s<span class="ot">=</span><span class="fu">sum</span>(mat.exp[i,])</span>
<span id="cb58-22"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-22" tabindex="-1"></a>                         p.o<span class="ot">=</span><span class="fu">sum</span>(p1.s,p2.s)</span>
<span id="cb58-23"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-23" tabindex="-1"></a>                         <span class="cf">if</span>(SKrows[i]<span class="sc">==</span>T){</span>
<span id="cb58-24"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-24" tabindex="-1"></a>                           test<span class="ot">=</span><span class="fu">twobinom</span>(<span class="at">r1=</span>p1.s,<span class="at">n1=</span>p.o,</span>
<span id="cb58-25"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-25" tabindex="-1"></a>                                         <span class="at">r2=</span>p2.s,<span class="at">n2=</span>p.o)<span class="sc">$</span>p.value</span>
<span id="cb58-26"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-26" tabindex="-1"></a>                         }<span class="cf">else</span>{</span>
<span id="cb58-27"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-27" tabindex="-1"></a>                           test<span class="ot">=</span><span class="fu">fisher.test</span>(<span class="fu">matrix</span>(<span class="fu">c</span>(p1.s,p2.s,</span>
<span id="cb58-28"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-28" tabindex="-1"></a>                                                     p2.s,p1.s),<span class="at">ncol =</span> <span class="dv">2</span>))<span class="sc">$</span>p.value</span>
<span id="cb58-29"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-29" tabindex="-1"></a>                         }</span>
<span id="cb58-30"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-30" tabindex="-1"></a>                         return.append<span class="ot">=</span><span class="fu">data.frame</span>(<span class="at">SNP_gene=</span>SNP_gene,<span class="at">p=</span>test)</span>
<span id="cb58-31"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-31" tabindex="-1"></a>                         <span class="fu">return</span>(return.append)</span>
<span id="cb58-32"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-32" tabindex="-1"></a>                       }</span>
<span id="cb58-33"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-33" tabindex="-1"></a>  return.df<span class="ot">=</span>return.df[<span class="fu">match</span>(<span class="fu">row.names</span>(pat.exp), return.df<span class="sc">$</span>SNP_gene),]</span>
<span id="cb58-34"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-34" tabindex="-1"></a>  </span>
<span id="cb58-35"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-35" tabindex="-1"></a>  <span class="co"># Return</span></span>
<span id="cb58-36"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-36" tabindex="-1"></a>  <span class="fu">return</span>(return.df)</span>
<span id="cb58-37"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb58-37" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb59-1" tabindex="-1"></a><span class="co"># Execute</span></span>
<span id="cb59-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb59-2" tabindex="-1"></a>WL.SK <span class="ot">&lt;-</span> <span class="fu">AST.SK</span>(WL_counts_normalized,<span class="st">&quot;WL&quot;</span>,<span class="dv">6</span>)</span>
<span id="cb59-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb59-3" tabindex="-1"></a>QL.SK <span class="ot">&lt;-</span> <span class="fu">AST.SK</span>(QL_counts_normalized,<span class="st">&quot;QL&quot;</span>,<span class="dv">6</span>)</span>
<span id="cb59-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb59-4" tabindex="-1"></a></span>
<span id="cb59-5"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb59-5" tabindex="-1"></a><span class="co"># Save output</span></span>
<span id="cb59-6"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb59-6" tabindex="-1"></a><span class="fu">write.csv</span>(WL.SK,<span class="st">&quot;AST-tutorial/ANALYSIS/RESULTS/WLSK.csv&quot;</span>, <span class="at">row.names=</span>F)</span>
<span id="cb59-7"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb59-7" tabindex="-1"></a><span class="fu">write.csv</span>(QL.SK,<span class="st">&quot;AST-tutorial/ANALYSIS/RESULTS/QLSK.csv&quot;</span>, <span class="at">row.names=</span>F)</span></code></pre></div>
</div>
</div>
<div id="gene-level-general-linear-mixed-models-glimmix" class="section level3 hasAnchor" number="3.5.2">
<h3><span class="header-section-number">3.5.2</span> Gene-level general linear mixed models (GLIMMIX)<a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#gene-level-general-linear-mixed-models-glimmix" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>A general linear mixed-effects model with interaction terms (GLIMMIX) is fit for each gene to assess the effects of Parent, Lineage, and their interaction on the raw read counts at each SNP, using the log of the library size factors as an offset to adjust for variation in sequencing depth between libraries.</p>
<div id="function-ast.glimmix" class="section level5 unnumbered hasAnchor">
<h5>Function: <code>AST.GLIMMIX</code><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#function-ast.glimmix" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>This is a wrapper function to fit a GLIMMIX for each gene in the count matrix and perform Wald tests to assess the statistical significance of the effects (parent, lineage, and their interaction). Wald test <em>p</em>-values are reported for each effect for each gene. Genes for which a model could not be fit due to errors are returned with a <em>p</em>-value of “1” for each effect.</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-1" tabindex="-1"></a>AST.GLIMMIX <span class="ot">&lt;-</span> <span class="cf">function</span>(counts,size_factors,cores){</span>
<span id="cb60-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-2" tabindex="-1"></a>  <span class="co"># counts = phenotype-specific counts matrix</span></span>
<span id="cb60-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-3" tabindex="-1"></a>  <span class="co"># phenotype = corresponding phenotype of counts</span></span>
<span id="cb60-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-4" tabindex="-1"></a>  <span class="co"># cores = # of threads for multithreaded search</span></span>
<span id="cb60-5"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-5" tabindex="-1"></a>  <span class="co"># size_factors = object from calcSizeFactors</span></span>
<span id="cb60-6"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-6" tabindex="-1"></a>  </span>
<span id="cb60-7"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-7" tabindex="-1"></a>  sizeFactors <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(size_factors[<span class="fu">names</span>(size_factors)<span class="sc">%in%</span><span class="fu">names</span>(counts)]))</span>
<span id="cb60-8"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-8" tabindex="-1"></a>  sizeFactors <span class="ot">&lt;-</span> <span class="fu">gather</span>(sizeFactors, sample.id, sizeFactor, </span>
<span id="cb60-9"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-9" tabindex="-1"></a>                        <span class="fu">names</span>(sizeFactors),<span class="at">factor_key=</span><span class="cn">FALSE</span>)</span>
<span id="cb60-10"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-10" tabindex="-1"></a>  counts<span class="sc">$</span>SNP_gene <span class="ot">&lt;-</span> <span class="fu">row.names</span>(counts)</span>
<span id="cb60-11"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-11" tabindex="-1"></a>  counts<span class="sc">$</span>geneID <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">unlist</span>(<span class="fu">map</span>(<span class="fu">strsplit</span>(counts<span class="sc">$</span>SNP_gene, </span>
<span id="cb60-12"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-12" tabindex="-1"></a>                                                    <span class="at">split =</span> <span class="st">&quot;:&quot;</span>), <span class="dv">2</span>)))</span>
<span id="cb60-13"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-13" tabindex="-1"></a>  genelist <span class="ot">&lt;-</span> <span class="fu">unique</span>(counts<span class="sc">$</span>geneID)</span>
<span id="cb60-14"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-14" tabindex="-1"></a>  <span class="fu">registerDoParallel</span>(<span class="at">cores=</span>cores)</span>
<span id="cb60-15"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-15" tabindex="-1"></a>  i.len <span class="ot">&lt;-</span> <span class="fu">length</span>(genelist)</span>
<span id="cb60-16"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-16" tabindex="-1"></a>  </span>
<span id="cb60-17"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-17" tabindex="-1"></a>  df.out <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i=</span><span class="dv">1</span><span class="sc">:</span>i.len,<span class="at">.combine=</span>rbind) <span class="sc">%dopar%</span> {</span>
<span id="cb60-18"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-18" tabindex="-1"></a>    counts.sub <span class="ot">&lt;-</span> counts[counts<span class="sc">$</span>geneID<span class="sc">==</span>genelist[i],]</span>
<span id="cb60-19"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-19" tabindex="-1"></a>    counts.sub<span class="sc">$</span>geneID <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb60-20"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-20" tabindex="-1"></a>    counts.sub <span class="ot">&lt;-</span> <span class="fu">gather</span>(counts.sub, sample.id, count, </span>
<span id="cb60-21"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-21" tabindex="-1"></a>                         <span class="fu">names</span>(counts.sub), <span class="sc">-</span>SNP_gene, <span class="at">factor_key=</span><span class="cn">TRUE</span>)</span>
<span id="cb60-22"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-22" tabindex="-1"></a>    counts.sub <span class="ot">&lt;-</span> <span class="fu">join</span>(counts.sub, metadata, <span class="at">by =</span> <span class="st">&quot;sample.id&quot;</span>)</span>
<span id="cb60-23"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-23" tabindex="-1"></a>    counts.sub <span class="ot">&lt;-</span> <span class="fu">join</span>(counts.sub,sizeFactors,<span class="at">by=</span><span class="st">&quot;sample.id&quot;</span>)</span>
<span id="cb60-24"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-24" tabindex="-1"></a>    counts.sub<span class="sc">$</span>parent <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(<span class="fu">str_sub</span>(counts.sub<span class="sc">$</span>parent,<span class="sc">-</span><span class="dv">1</span>,<span class="sc">-</span><span class="dv">1</span>))</span>
<span id="cb60-25"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-25" tabindex="-1"></a>    counts.sub<span class="sc">$</span>SNP_gene <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(counts.sub<span class="sc">$</span>SNP_gene)</span>
<span id="cb60-26"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-26" tabindex="-1"></a>    counts.sub<span class="sc">$</span>lineage <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(counts.sub<span class="sc">$</span>lineage)</span>
<span id="cb60-27"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-27" tabindex="-1"></a>    counts.sub<span class="sc">$</span>individual <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(counts.sub<span class="sc">$</span>individual)</span>
<span id="cb60-28"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-28" tabindex="-1"></a>    testfail <span class="ot">&lt;-</span> F</span>
<span id="cb60-29"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-29" tabindex="-1"></a>    test <span class="ot">&lt;-</span> <span class="st">&quot;null&quot;</span></span>
<span id="cb60-30"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-30" tabindex="-1"></a>    <span class="fu">tryCatchLog</span>(test <span class="ot">&lt;-</span> <span class="fu">lmer</span>(count<span class="sc">~</span>parent<span class="sc">+</span>lineage<span class="sc">+</span>parent<span class="sc">*</span>lineage<span class="sc">+</span>(<span class="dv">1</span><span class="sc">|</span>SNP_gene)<span class="sc">+</span></span>
<span id="cb60-31"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-31" tabindex="-1"></a>                               (<span class="dv">1</span><span class="sc">|</span>individual)<span class="sc">+</span><span class="fu">offset</span>(<span class="fu">log</span>(sizeFactor)),<span class="at">data=</span>counts.sub), </span>
<span id="cb60-32"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-32" tabindex="-1"></a>                <span class="at">error =</span> <span class="cf">function</span>(e) {testfail <span class="ot">&lt;-</span> T})</span>
<span id="cb60-33"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-33" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">class</span>(test)<span class="sc">==</span><span class="st">&quot;character&quot;</span>){testfail <span class="ot">&lt;-</span> T}</span>
<span id="cb60-34"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-34" tabindex="-1"></a>    <span class="cf">if</span>(testfail<span class="sc">==</span>F){</span>
<span id="cb60-35"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-35" tabindex="-1"></a>      test <span class="ot">&lt;-</span> <span class="fu">summary</span>(test)</span>
<span id="cb60-36"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-36" tabindex="-1"></a>      parent.p.list <span class="ot">&lt;-</span> test[[<span class="st">&quot;coefficients&quot;</span>]][<span class="dv">2</span>,<span class="dv">5</span>]</span>
<span id="cb60-37"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-37" tabindex="-1"></a>      Lineage.p.list <span class="ot">&lt;-</span> test[[<span class="st">&quot;coefficients&quot;</span>]][<span class="dv">3</span>,<span class="dv">5</span>]</span>
<span id="cb60-38"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-38" tabindex="-1"></a>      parent.Lineage.p.list <span class="ot">&lt;-</span> test[[<span class="st">&quot;coefficients&quot;</span>]][<span class="dv">4</span>,<span class="dv">5</span>]</span>
<span id="cb60-39"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-39" tabindex="-1"></a>    }<span class="cf">else</span>{</span>
<span id="cb60-40"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-40" tabindex="-1"></a>      parent.p.list <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb60-41"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-41" tabindex="-1"></a>      Lineage.p.list <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb60-42"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-42" tabindex="-1"></a>      parent.Lineage.p.list <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb60-43"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-43" tabindex="-1"></a>    }</span>
<span id="cb60-44"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-44" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">data.frame</span>(<span class="at">ID=</span>genelist[i],</span>
<span id="cb60-45"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-45" tabindex="-1"></a>                      <span class="at">parent.p=</span>parent.p.list,</span>
<span id="cb60-46"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-46" tabindex="-1"></a>                      <span class="at">Lineage.p=</span>Lineage.p.list,</span>
<span id="cb60-47"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-47" tabindex="-1"></a>                      <span class="at">parentXLineage.p=</span>parent.Lineage.p.list))</span>
<span id="cb60-48"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-48" tabindex="-1"></a>  }</span>
<span id="cb60-49"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-49" tabindex="-1"></a>  <span class="fu">return</span>(df.out)</span>
<span id="cb60-50"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb60-50" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb61-1" tabindex="-1"></a><span class="co"># Execute</span></span>
<span id="cb61-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb61-2" tabindex="-1"></a>WL.GLIMMIX <span class="ot">&lt;-</span> <span class="fu">AST.GLIMMIX</span>(WL_counts,size_factors,<span class="dv">6</span>)</span>
<span id="cb61-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb61-3" tabindex="-1"></a>QL.GLIMMIX <span class="ot">&lt;-</span> <span class="fu">AST.GLIMMIX</span>(QL_counts,size_factors,<span class="dv">6</span>)</span>
<span id="cb61-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb61-4" tabindex="-1"></a></span>
<span id="cb61-5"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb61-5" tabindex="-1"></a><span class="co"># Save output</span></span>
<span id="cb61-6"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb61-6" tabindex="-1"></a><span class="fu">write.csv</span>(WL.GLIMMIX,<span class="st">&quot;AST-tutorial/ANALYSIS/RESULTS/WLGLIMMIX.csv&quot;</span>, <span class="at">row.names=</span>F)</span>
<span id="cb61-7"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb61-7" tabindex="-1"></a><span class="fu">write.csv</span>(QL.GLIMMIX,<span class="st">&quot;AST-tutorial/ANALYSIS/RESULTS/QLGLIMMIX.csv&quot;</span>, <span class="at">row.names=</span>F)</span></code></pre></div>
</div>
</div>
</div>
<div id="allele-specific-transcription-analysis" class="section level2 hasAnchor" number="3.6">
<h2><span class="header-section-number">3.6</span> Allele-specific transcription analysis<a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#allele-specific-transcription-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="fdr-correction-and-threshold-tests" class="section level3 hasAnchor" number="3.6.1">
<h3><span class="header-section-number">3.6.1</span> FDR correction and threshold tests<a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#fdr-correction-and-threshold-tests" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The False Discovery Rate (FDR) is now computed for the Storer-Kim and GLIMMIX (Wald) test results, requiring FDR &lt; 0.05 for significance. For a gene to be considered as showing parent-of-origin or lineage-of-origin effects, all SNPs are required to exhibit the same directional bias (i.e., maternal or paternal) in the Storer-Kim test, and a strictly parent-of-origin or lineage-of-origin effect in the GLIMMIX. To avoid identifying genes with parent-of-origin effects influenced by lineage-of-origin effects, or lineage-of-origin genes influenced by parent-of-origin effects, genes with a significant interaction effect are considered unbiased.</p>
<p>Additionally, the proportions of Lineage A reads in samples with a Lineage B mother and Lineage A father (proportion 1, or <em>p1</em>), and the proportion of Lineage A reads in samples with a Lineage A mother and Lineage B father (<em>p2</em>), are subjected to a threshold test following methods described in <a href="https://doi.org/10.1038/hdy.2014.18">Wang &amp; Clark, 2014, <em>Heredity</em></a>. Specifically, thresholds of <em>p1</em>&lt;0.4 and <em>p2</em>&gt;0.6 are required for maternal bias, <em>p1</em>&gt;0.6 and <em>p2</em>&lt;0.4 for paternal bias, <em>p1</em>&lt;0.4 and <em>p2</em>&lt;0.4 for lineage B bias, and <em>p1</em>&gt;0.6 and <em>p2</em>&gt;0.6 for lineage A bias.</p>
<div id="function-ast.analysis" class="section level5 unnumbered hasAnchor">
<h5>Function: <code>AST.Analysis</code><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#function-ast.analysis" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-1" tabindex="-1"></a>AST.Analysis <span class="ot">&lt;-</span> <span class="cf">function</span>(counts,phenotype,SK,GLIMMIX){</span>
<span id="cb62-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-2" tabindex="-1"></a>  <span class="co"># counts = phenotype-specific count matrix</span></span>
<span id="cb62-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-3" tabindex="-1"></a>  <span class="co"># phenotype = corresponding phenotype of counts</span></span>
<span id="cb62-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-4" tabindex="-1"></a>  <span class="co"># SK = object from AST.SK</span></span>
<span id="cb62-5"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-5" tabindex="-1"></a>  <span class="co"># GLIMMIX = object from AST.GLIMMIX</span></span>
<span id="cb62-6"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-6" tabindex="-1"></a>  </span>
<span id="cb62-7"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-7" tabindex="-1"></a>  <span class="co"># Split count matrices by lineage and parent for plotting</span></span>
<span id="cb62-8"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-8" tabindex="-1"></a>  counts <span class="ot">&lt;-</span> counts[,<span class="fu">names</span>(counts)<span class="sc">%in%</span>metadata<span class="sc">$</span>sample.id]</span>
<span id="cb62-9"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-9" tabindex="-1"></a>  p1.pat <span class="ot">&lt;-</span> counts[,metadata[metadata<span class="sc">$</span>parent<span class="sc">%in%</span><span class="fu">c</span>(<span class="st">&quot;M&quot;</span>)<span class="sc">&amp;</span>metadata<span class="sc">$</span>lineage<span class="sc">==</span><span class="st">&quot;B&quot;</span><span class="sc">&amp;</span>metadata<span class="sc">$</span>phenotype<span class="sc">==</span>phenotype,<span class="st">&quot;sample.id&quot;</span>]]</span>
<span id="cb62-10"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-10" tabindex="-1"></a>  p1.mat <span class="ot">&lt;-</span> counts[,metadata[metadata<span class="sc">$</span>parent<span class="sc">%in%</span><span class="fu">c</span>(<span class="st">&quot;F&quot;</span>)<span class="sc">&amp;</span>metadata<span class="sc">$</span>lineage<span class="sc">==</span><span class="st">&quot;B&quot;</span><span class="sc">&amp;</span>metadata<span class="sc">$</span>phenotype<span class="sc">==</span>phenotype,<span class="st">&quot;sample.id&quot;</span>]]</span>
<span id="cb62-11"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-11" tabindex="-1"></a>  p2.pat <span class="ot">&lt;-</span> counts[,metadata[metadata<span class="sc">$</span>parent<span class="sc">%in%</span><span class="fu">c</span>(<span class="st">&quot;M&quot;</span>)<span class="sc">&amp;</span>metadata<span class="sc">$</span>lineage<span class="sc">==</span><span class="st">&quot;A&quot;</span><span class="sc">&amp;</span>metadata<span class="sc">$</span>phenotype<span class="sc">==</span>phenotype,<span class="st">&quot;sample.id&quot;</span>]]</span>
<span id="cb62-12"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-12" tabindex="-1"></a>  p2.mat <span class="ot">&lt;-</span> counts[,metadata[metadata<span class="sc">$</span>parent<span class="sc">%in%</span><span class="fu">c</span>(<span class="st">&quot;F&quot;</span>)<span class="sc">&amp;</span>metadata<span class="sc">$</span>lineage<span class="sc">==</span><span class="st">&quot;A&quot;</span><span class="sc">&amp;</span>metadata<span class="sc">$</span>phenotype<span class="sc">==</span>phenotype,<span class="st">&quot;sample.id&quot;</span>]]</span>
<span id="cb62-13"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-13" tabindex="-1"></a>  </span>
<span id="cb62-14"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-14" tabindex="-1"></a>  <span class="co"># Set up a data.frame to plot p1 and p2 for each SNP</span></span>
<span id="cb62-15"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-15" tabindex="-1"></a>  p1.plot <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">rowSums</span>(p1.pat)<span class="sc">/</span>(<span class="fu">rowSums</span>(p1.mat)<span class="sc">+</span><span class="fu">rowSums</span>(p1.pat)))</span>
<span id="cb62-16"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-16" tabindex="-1"></a>  <span class="fu">names</span>(p1.plot) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;p1&quot;</span>)</span>
<span id="cb62-17"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-17" tabindex="-1"></a>  p1.plot[<span class="fu">is.nan</span>(p1.plot<span class="sc">$</span>p1),<span class="st">&quot;p1&quot;</span>] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb62-18"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-18" tabindex="-1"></a>  p2.plot <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">rowSums</span>(p2.mat)<span class="sc">/</span>(<span class="fu">rowSums</span>(p2.mat)<span class="sc">+</span><span class="fu">rowSums</span>(p2.pat)))</span>
<span id="cb62-19"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-19" tabindex="-1"></a>  <span class="fu">names</span>(p2.plot) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;p2&quot;</span>)</span>
<span id="cb62-20"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-20" tabindex="-1"></a>  p2.plot[<span class="fu">is.nan</span>(p2.plot<span class="sc">$</span>p2),<span class="st">&quot;p2&quot;</span>] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb62-21"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-21" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> <span class="fu">cbind</span>(p1.plot,p2.plot)</span>
<span id="cb62-22"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-22" tabindex="-1"></a>  </span>
<span id="cb62-23"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-23" tabindex="-1"></a>  <span class="co"># Join results of Storer-Kim tests</span></span>
<span id="cb62-24"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-24" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> plot[<span class="fu">row.names</span>(plot)<span class="sc">%in%</span>SK<span class="sc">$</span>SNP_gene,]</span>
<span id="cb62-25"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-25" tabindex="-1"></a>  plot<span class="sc">$</span>SK.p <span class="ot">&lt;-</span> SK<span class="sc">$</span>p</span>
<span id="cb62-26"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-26" tabindex="-1"></a>  plot<span class="sc">$</span>SNP_gene <span class="ot">&lt;-</span> <span class="fu">row.names</span>(plot)</span>
<span id="cb62-27"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-27" tabindex="-1"></a>  plot<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">map</span>(<span class="fu">strsplit</span>(plot<span class="sc">$</span>SNP_gene, <span class="at">split =</span> <span class="st">&quot;:&quot;</span>), <span class="dv">2</span>))</span>
<span id="cb62-28"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-28" tabindex="-1"></a>  </span>
<span id="cb62-29"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-29" tabindex="-1"></a>  <span class="co"># Reformat output from GLIMMMIX models</span></span>
<span id="cb62-30"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-30" tabindex="-1"></a>  GLIMMIX.biased <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene=</span>GLIMMIX<span class="sc">$</span>ID,</span>
<span id="cb62-31"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-31" tabindex="-1"></a>                               <span class="at">parent.p=</span>GLIMMIX<span class="sc">$</span>parent.p,</span>
<span id="cb62-32"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-32" tabindex="-1"></a>                               <span class="at">Lineage.p=</span>GLIMMIX<span class="sc">$</span>Lineage.p,</span>
<span id="cb62-33"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-33" tabindex="-1"></a>                               <span class="at">parentXLineage.p=</span>GLIMMIX<span class="sc">$</span>parentXLineage.p)</span>
<span id="cb62-34"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-34" tabindex="-1"></a>  </span>
<span id="cb62-35"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-35" tabindex="-1"></a>  <span class="co"># Correct for multiple testing</span></span>
<span id="cb62-36"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-36" tabindex="-1"></a>  plot<span class="sc">$</span>SK.padj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(plot<span class="sc">$</span>SK.p,<span class="st">&quot;BH&quot;</span>)</span>
<span id="cb62-37"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-37" tabindex="-1"></a>  plot<span class="sc">$</span>bias <span class="ot">&lt;-</span> <span class="st">&quot;NA&quot;</span></span>
<span id="cb62-38"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-38" tabindex="-1"></a>  GLIMMIX<span class="sc">$</span>parent.padj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(GLIMMIX<span class="sc">$</span>parent.p,<span class="st">&quot;BH&quot;</span>)</span>
<span id="cb62-39"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-39" tabindex="-1"></a>  GLIMMIX<span class="sc">$</span>Lineage.padj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(GLIMMIX<span class="sc">$</span>Lineage.p,<span class="st">&quot;BH&quot;</span>)</span>
<span id="cb62-40"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-40" tabindex="-1"></a>  GLIMMIX<span class="sc">$</span>parentXLineage.padj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(GLIMMIX<span class="sc">$</span>parentXLineage.p,<span class="st">&quot;BH&quot;</span>)</span>
<span id="cb62-41"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-41" tabindex="-1"></a>  GLIMMIX.biased <span class="ot">&lt;-</span> GLIMMIX[GLIMMIX<span class="sc">$</span>parent.padj<span class="sc">&lt;</span><span class="fl">0.05</span><span class="sc">|</span>GLIMMIX<span class="sc">$</span>Lineage.padj<span class="sc">&lt;</span><span class="fl">0.05</span>,<span class="dv">1</span>]</span>
<span id="cb62-42"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-42" tabindex="-1"></a>  GLIMMIX.biased <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(GLIMMIX.biased,GLIMMIX[GLIMMIX<span class="sc">$</span>parentXLineage.padj<span class="sc">&lt;</span><span class="fl">0.05</span>,<span class="dv">1</span>])</span>
<span id="cb62-43"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-43" tabindex="-1"></a>  </span>
<span id="cb62-44"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-44" tabindex="-1"></a>  <span class="co"># For each gene, check whether all SNPs are biased in the same direction at established thresholds</span></span>
<span id="cb62-45"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-45" tabindex="-1"></a>  <span class="do">## Genes with parentXLineage effects are flagged as unbiased</span></span>
<span id="cb62-46"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-46" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">row.names</span>(plot))){</span>
<span id="cb62-47"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-47" tabindex="-1"></a>    p <span class="ot">&lt;-</span> plot[i,<span class="st">&quot;SK.padj&quot;</span>]</span>
<span id="cb62-48"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-48" tabindex="-1"></a>    p1 <span class="ot">&lt;-</span> plot[i,<span class="st">&quot;p1&quot;</span>]</span>
<span id="cb62-49"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-49" tabindex="-1"></a>    p2 <span class="ot">&lt;-</span> plot[i,<span class="st">&quot;p2&quot;</span>]</span>
<span id="cb62-50"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-50" tabindex="-1"></a>    <span class="cf">if</span>(p<span class="sc">&lt;</span><span class="fl">0.05</span><span class="sc">&amp;</span>p1<span class="sc">&gt;</span><span class="fl">0.6</span><span class="sc">&amp;</span>p2<span class="sc">&lt;</span><span class="fl">0.4</span>){plot[i,<span class="st">&quot;bias&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;pat&quot;</span>}</span>
<span id="cb62-51"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-51" tabindex="-1"></a>    <span class="cf">if</span>(p<span class="sc">&lt;</span><span class="fl">0.05</span><span class="sc">&amp;</span>p1<span class="sc">&lt;</span><span class="fl">0.4</span><span class="sc">&amp;</span>p2<span class="sc">&gt;</span><span class="fl">0.6</span>){plot[i,<span class="st">&quot;bias&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;mat&quot;</span>}</span>
<span id="cb62-52"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-52" tabindex="-1"></a>    <span class="cf">if</span>(p<span class="sc">&lt;</span><span class="fl">0.05</span><span class="sc">&amp;</span>p1<span class="sc">&lt;</span><span class="fl">0.4</span><span class="sc">&amp;</span>p2<span class="sc">&lt;</span><span class="fl">0.4</span>){plot[i,<span class="st">&quot;bias&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Lineage B&quot;</span>}</span>
<span id="cb62-53"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-53" tabindex="-1"></a>    <span class="cf">if</span>(p<span class="sc">&lt;</span><span class="fl">0.05</span><span class="sc">&amp;</span>p1<span class="sc">&gt;</span><span class="fl">0.6</span><span class="sc">&amp;</span>p2<span class="sc">&gt;</span><span class="fl">0.6</span>){plot[i,<span class="st">&quot;bias&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Lineage A&quot;</span>}</span>
<span id="cb62-54"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-54" tabindex="-1"></a>  }</span>
<span id="cb62-55"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-55" tabindex="-1"></a>  biaslist <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">matrix</span>(<span class="at">ncol=</span><span class="dv">2</span>,<span class="at">nrow=</span><span class="dv">0</span>))</span>
<span id="cb62-56"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-56" tabindex="-1"></a>  <span class="fu">names</span>(biaslist) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;gene&quot;</span>,<span class="st">&quot;bias&quot;</span>)</span>
<span id="cb62-57"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-57" tabindex="-1"></a>  genelist <span class="ot">&lt;-</span> <span class="fu">unique</span>(plot<span class="sc">$</span>gene)</span>
<span id="cb62-58"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-58" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(genelist)){</span>
<span id="cb62-59"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-59" tabindex="-1"></a>    tmp <span class="ot">&lt;-</span> <span class="fu">unique</span>(plot[plot<span class="sc">$</span>gene<span class="sc">==</span>genelist[i],<span class="st">&quot;bias&quot;</span>])</span>
<span id="cb62-60"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-60" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(tmp)<span class="sc">&gt;</span><span class="dv">1</span>){</span>
<span id="cb62-61"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-61" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">length</span>(tmp)<span class="sc">==</span><span class="dv">2</span>){</span>
<span id="cb62-62"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-62" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">any</span>(tmp<span class="sc">%in%</span><span class="st">&quot;NA&quot;</span>)){</span>
<span id="cb62-63"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-63" tabindex="-1"></a>          bias <span class="ot">&lt;-</span> tmp[<span class="sc">!</span>tmp<span class="sc">%in%</span><span class="st">&quot;NA&quot;</span>]</span>
<span id="cb62-64"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-64" tabindex="-1"></a>        }<span class="cf">else</span>{bias <span class="ot">&lt;-</span> <span class="st">&quot;NA&quot;</span>}</span>
<span id="cb62-65"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-65" tabindex="-1"></a>      }<span class="cf">else</span>{</span>
<span id="cb62-66"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-66" tabindex="-1"></a>        bias <span class="ot">&lt;-</span> <span class="st">&quot;NA&quot;</span></span>
<span id="cb62-67"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-67" tabindex="-1"></a>      }</span>
<span id="cb62-68"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-68" tabindex="-1"></a>    }<span class="cf">else</span>{bias <span class="ot">&lt;-</span> tmp}</span>
<span id="cb62-69"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-69" tabindex="-1"></a>    biaslist <span class="ot">&lt;-</span> <span class="fu">rbind</span>(biaslist,<span class="fu">data.frame</span>(<span class="at">gene=</span>genelist[[i]], <span class="at">bias=</span>bias))</span>
<span id="cb62-70"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-70" tabindex="-1"></a>  }</span>
<span id="cb62-71"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-71" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> plot <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(biaslist, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&#39;gene&#39;</span> <span class="ot">=</span> <span class="st">&#39;gene&#39;</span>)) </span>
<span id="cb62-72"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-72" tabindex="-1"></a>  <span class="fu">names</span>(plot)[<span class="fu">c</span>(<span class="dv">7</span><span class="sc">:</span><span class="dv">8</span>)] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;xbias&quot;</span>,<span class="st">&quot;bias&quot;</span>)</span>
<span id="cb62-73"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-73" tabindex="-1"></a>  plot<span class="sc">$</span>bias.plot <span class="ot">&lt;-</span> <span class="st">&quot;NA&quot;</span></span>
<span id="cb62-74"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-74" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">row.names</span>(plot))){</span>
<span id="cb62-75"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-75" tabindex="-1"></a>    p1 <span class="ot">&lt;-</span> plot<span class="sc">$</span>p1[i]</span>
<span id="cb62-76"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-76" tabindex="-1"></a>    p2 <span class="ot">&lt;-</span> plot<span class="sc">$</span>p2[i]</span>
<span id="cb62-77"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-77" tabindex="-1"></a>    bias <span class="ot">&lt;-</span> plot<span class="sc">$</span>bias[i]</span>
<span id="cb62-78"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-78" tabindex="-1"></a>    <span class="cf">if</span>(<span class="sc">!</span>bias<span class="sc">==</span><span class="st">&quot;NA&quot;</span>){</span>
<span id="cb62-79"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-79" tabindex="-1"></a>      <span class="cf">if</span>(bias<span class="sc">==</span><span class="st">&quot;pat&quot;</span>){<span class="cf">if</span>(p1<span class="sc">&gt;</span><span class="fl">0.6</span><span class="sc">&amp;</span>p2<span class="sc">&lt;</span><span class="fl">0.4</span>){plot[i,<span class="st">&quot;bias.plot&quot;</span>]<span class="ot">&lt;-</span> <span class="st">&quot;pat&quot;</span>}}</span>
<span id="cb62-80"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-80" tabindex="-1"></a>      <span class="cf">if</span>(bias<span class="sc">==</span><span class="st">&quot;mat&quot;</span>){<span class="cf">if</span>(p1<span class="sc">&lt;</span><span class="fl">0.4</span><span class="sc">&amp;</span>p2<span class="sc">&gt;</span><span class="fl">0.6</span>){plot[i,<span class="st">&quot;bias.plot&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;mat&quot;</span>}}</span>
<span id="cb62-81"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-81" tabindex="-1"></a>      <span class="cf">if</span>(bias<span class="sc">==</span><span class="st">&quot;Lineage B&quot;</span>){<span class="cf">if</span>(p1<span class="sc">&lt;</span><span class="fl">0.4</span><span class="sc">&amp;</span>p2<span class="sc">&lt;</span><span class="fl">0.4</span>){plot[i,<span class="st">&quot;bias.plot&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Lineage B&quot;</span>}}</span>
<span id="cb62-82"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-82" tabindex="-1"></a>      <span class="cf">if</span>(bias<span class="sc">==</span><span class="st">&quot;Lineage A&quot;</span>){<span class="cf">if</span>(p1<span class="sc">&gt;</span><span class="fl">0.6</span><span class="sc">&amp;</span>p2<span class="sc">&gt;</span><span class="fl">0.6</span>){plot[i,<span class="st">&quot;bias.plot&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Lineage A&quot;</span>}}</span>
<span id="cb62-83"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-83" tabindex="-1"></a>    }</span>
<span id="cb62-84"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-84" tabindex="-1"></a>  }</span>
<span id="cb62-85"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-85" tabindex="-1"></a>  plot[<span class="sc">!</span>plot<span class="sc">$</span>gene<span class="sc">%in%</span>GLIMMIX.biased,<span class="st">&quot;bias.plot&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;NA&quot;</span> </span>
<span id="cb62-86"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-86" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> <span class="fu">rbind</span>(plot[plot<span class="sc">$</span>bias.plot<span class="sc">%in%</span><span class="fu">c</span>(<span class="st">&quot;NA&quot;</span>),],</span>
<span id="cb62-87"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-87" tabindex="-1"></a>                plot[plot<span class="sc">$</span>bias.plot<span class="sc">%in%</span><span class="fu">c</span>(<span class="st">&quot;mat&quot;</span>, <span class="st">&quot;Lineage A&quot;</span>, <span class="st">&quot;Lineage B&quot;</span>, <span class="st">&quot;pat&quot;</span>),])</span>
<span id="cb62-88"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-88" tabindex="-1"></a>  plot<span class="sc">$</span>bias.plot <span class="ot">&lt;-</span> <span class="fu">factor</span>(plot<span class="sc">$</span>bias.plot,</span>
<span id="cb62-89"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-89" tabindex="-1"></a>                           <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;NA&quot;</span>,<span class="st">&quot;mat&quot;</span>, <span class="st">&quot;Lineage A&quot;</span>, <span class="st">&quot;Lineage B&quot;</span>, <span class="st">&quot;pat&quot;</span>))</span>
<span id="cb62-90"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-90" tabindex="-1"></a>  </span>
<span id="cb62-91"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-91" tabindex="-1"></a>  <span class="co"># Return</span></span>
<span id="cb62-92"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-92" tabindex="-1"></a>  <span class="fu">return</span>(plot)</span>
<span id="cb62-93"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb62-93" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb63-1" tabindex="-1"></a><span class="co"># Execute</span></span>
<span id="cb63-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb63-2" tabindex="-1"></a>WL.plot <span class="ot">&lt;-</span> <span class="fu">AST.Analysis</span>(WL_counts_normalized,</span>
<span id="cb63-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb63-3" tabindex="-1"></a>                        <span class="st">&quot;WL&quot;</span>,metadata,WL.SK,WL.GLIMMIX)</span>
<span id="cb63-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb63-4" tabindex="-1"></a>QL.plot <span class="ot">&lt;-</span> <span class="fu">AST.Analysis</span>(QL_counts_normalized,</span>
<span id="cb63-5"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb63-5" tabindex="-1"></a>                        <span class="st">&quot;QL&quot;</span>,metadata,QL.SK,QL.GLIMMIX)</span>
<span id="cb63-6"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb63-6" tabindex="-1"></a></span>
<span id="cb63-7"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb63-7" tabindex="-1"></a><span class="co"># Save output</span></span>
<span id="cb63-8"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb63-8" tabindex="-1"></a><span class="fu">write.csv</span>(WL.plot,<span class="st">&quot;AST-tutorial/ANALYSIS/RESULTS/WLAST.csv&quot;</span>,<span class="at">row.names=</span>F)</span>
<span id="cb63-9"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb63-9" tabindex="-1"></a><span class="fu">write.csv</span>(QL.plot,<span class="st">&quot;AST-tutorial/ANALYSIS/RESULTS/QLAST.csv&quot;</span>,<span class="at">row.names=</span>F)</span></code></pre></div>
<p>The last three columns of a dataframe produced by <code>AST.Analysis</code> are informative of SNP and gene-level allelic bias:</p>
<ul>
<li><code>xbias</code> = SNP-level bias given 1) threshold test applied to <code>p1</code> and <code>p2</code>, and 2) FDR-corrected SK-test <em>p</em>-value (<code>SK.padj</code>)</li>
<li><code>bias</code> = Gene-level bias given consistency of <code>xbias</code> value for each SNP</li>
<li><code>bias.plot</code> = Gene-level bias given <code>bias</code> and FDR-corrected GLIMMIX (Wald test) <em>p</em>-values. This is the reported value of allelic bias for each gene, and will be used to quantify the number of genes showing each category of allelic bias for performing Chi-squared tests.</li>
</ul>
</div>
</div>
<div id="chi-squared-tests" class="section level3 hasAnchor" number="3.6.2">
<h3><span class="header-section-number">3.6.2</span> Chi-squared tests<a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#chi-squared-tests" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Finally, allele-specific transcription is compared between phenotypes. Specifically, Chi-squared tests are performed for each category of allelic bias to test the hypothesis that the distributions of biased and unbiased genes in each phenotype are statistically different.</p>
<p>For these tests, we will use the honey bee gene set (saved as <code>${DIR_INDEX}/Amel_HAv3.1_geneIDs.txt</code> in Section 2) minus the genes in <code>filterlist</code>, as the “gene universe” - i.e., all possible genes that could have been expressed in these samples and tested for allele-specific transcription.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb64-1" tabindex="-1"></a>allgenes <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;AST-tutorial/INDEX/Amel_HAv3.1_geneIDs.txt&quot;</span>,<span class="at">header=</span>F)[,<span class="dv">1</span>]</span>
<span id="cb64-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb64-2" tabindex="-1"></a>allgenes <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(allgenes,filterlist)</span></code></pre></div>
<div id="function-ast.chisq" class="section level5 unnumbered hasAnchor">
<h5>Function: <code>AST.chisq</code><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#function-ast.chisq" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>This is a wrapper function to perform these tests and format a table for plotting.</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-1" tabindex="-1"></a>AST.chisq <span class="ot">&lt;-</span> <span class="cf">function</span>(pheno1.label,pheno1.plot,</span>
<span id="cb65-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-2" tabindex="-1"></a>                      pheno2.label,pheno2.plot,</span>
<span id="cb65-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-3" tabindex="-1"></a>                      allgenes){</span>
<span id="cb65-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-4" tabindex="-1"></a>  <span class="co"># pheno1.label = label for phenotype 1</span></span>
<span id="cb65-5"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-5" tabindex="-1"></a>  <span class="co"># pheno1.plot = object from AST.Analysis</span></span>
<span id="cb65-6"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-6" tabindex="-1"></a>  <span class="co"># pheno2.label = label for phenotype 2</span></span>
<span id="cb65-7"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-7" tabindex="-1"></a>  <span class="co"># pheno2.plot = object from AST.Analysis</span></span>
<span id="cb65-8"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-8" tabindex="-1"></a>  <span class="co"># allgenes = list containing background set of genes</span></span>
<span id="cb65-9"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-9" tabindex="-1"></a>  </span>
<span id="cb65-10"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-10" tabindex="-1"></a>  <span class="co"># Collapse pheno plots by gene</span></span>
<span id="cb65-11"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-11" tabindex="-1"></a>  pheno1.plot <span class="ot">&lt;-</span> pheno1.plot[,<span class="fu">c</span>(<span class="st">&quot;gene&quot;</span>,<span class="st">&quot;bias.plot&quot;</span>)]</span>
<span id="cb65-12"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-12" tabindex="-1"></a>  pheno1.plot <span class="ot">&lt;-</span> pheno1.plot[<span class="sc">!</span><span class="fu">duplicated</span>(pheno1.plot),]</span>
<span id="cb65-13"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-13" tabindex="-1"></a>  pheno2.plot <span class="ot">&lt;-</span> pheno2.plot[,<span class="fu">c</span>(<span class="st">&quot;gene&quot;</span>,<span class="st">&quot;bias.plot&quot;</span>)]</span>
<span id="cb65-14"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-14" tabindex="-1"></a>  pheno2.plot <span class="ot">&lt;-</span> pheno2.plot[<span class="sc">!</span><span class="fu">duplicated</span>(pheno2.plot),]</span>
<span id="cb65-15"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-15" tabindex="-1"></a>  </span>
<span id="cb65-16"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-16" tabindex="-1"></a>  <span class="co"># Quantify genes in each category of bias and combine into a table</span></span>
<span id="cb65-17"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-17" tabindex="-1"></a>  gmid.df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb65-18"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-18" tabindex="-1"></a>    <span class="at">Unresponsive=</span><span class="fu">c</span>(<span class="fu">length</span>(<span class="fu">unique</span>(pheno1.plot[pheno1.plot<span class="sc">$</span>bias.plot<span class="sc">==</span><span class="st">&quot;mat&quot;</span>,<span class="st">&quot;gene&quot;</span>])),</span>
<span id="cb65-19"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-19" tabindex="-1"></a>                   <span class="fu">length</span>(<span class="fu">unique</span>(pheno1.plot[pheno1.plot<span class="sc">$</span>bias.plot<span class="sc">==</span><span class="st">&quot;Lineage A&quot;</span>,<span class="st">&quot;gene&quot;</span>])),</span>
<span id="cb65-20"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-20" tabindex="-1"></a>                   <span class="fu">length</span>(<span class="fu">unique</span>(pheno1.plot[pheno1.plot<span class="sc">$</span>bias.plot<span class="sc">==</span><span class="st">&quot;Lineage B&quot;</span>,<span class="st">&quot;gene&quot;</span>])),</span>
<span id="cb65-21"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-21" tabindex="-1"></a>                   <span class="fu">length</span>(<span class="fu">unique</span>(pheno1.plot[pheno1.plot<span class="sc">$</span>bias.plot<span class="sc">==</span><span class="st">&quot;pat&quot;</span>,<span class="st">&quot;gene&quot;</span>]))),</span>
<span id="cb65-22"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-22" tabindex="-1"></a>    <span class="at">Bias=</span><span class="fu">c</span>(<span class="st">&quot;mat&quot;</span>,<span class="st">&quot;Lineage A&quot;</span>,<span class="st">&quot;Lineage B&quot;</span>,<span class="st">&quot;pat&quot;</span>),</span>
<span id="cb65-23"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-23" tabindex="-1"></a>    <span class="at">Responsive=</span><span class="fu">c</span>(<span class="fu">length</span>(<span class="fu">unique</span>(pheno2.plot[pheno2.plot<span class="sc">$</span>bias.plot<span class="sc">==</span><span class="st">&quot;mat&quot;</span>,<span class="st">&quot;gene&quot;</span>])),</span>
<span id="cb65-24"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-24" tabindex="-1"></a>                 <span class="fu">length</span>(<span class="fu">unique</span>(pheno2.plot[pheno2.plot<span class="sc">$</span>bias.plot<span class="sc">==</span><span class="st">&quot;Lineage A&quot;</span>,<span class="st">&quot;gene&quot;</span>])),</span>
<span id="cb65-25"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-25" tabindex="-1"></a>                 <span class="fu">length</span>(<span class="fu">unique</span>(pheno2.plot[pheno2.plot<span class="sc">$</span>bias.plot<span class="sc">==</span><span class="st">&quot;Lineage B&quot;</span>,<span class="st">&quot;gene&quot;</span>])),</span>
<span id="cb65-26"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-26" tabindex="-1"></a>                 <span class="fu">length</span>(<span class="fu">unique</span>(pheno2.plot[pheno2.plot<span class="sc">$</span>bias.plot<span class="sc">==</span><span class="st">&quot;pat&quot;</span>,<span class="st">&quot;gene&quot;</span>]))))</span>
<span id="cb65-27"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-27" tabindex="-1"></a>  </span>
<span id="cb65-28"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-28" tabindex="-1"></a>  <span class="co"># Perform Chi-squared tests</span></span>
<span id="cb65-29"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-29" tabindex="-1"></a>  mat.test <span class="ot">&lt;-</span> <span class="fu">chisq.test</span>(<span class="fu">data.frame</span>(<span class="at">Success=</span><span class="fu">c</span>(gmid.df[<span class="dv">1</span>,<span class="dv">1</span>],gmid.df[<span class="dv">1</span>,<span class="dv">3</span>]),</span>
<span id="cb65-30"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-30" tabindex="-1"></a>                                    <span class="at">Failure=</span><span class="fu">c</span>(<span class="fu">length</span>(allgenes)<span class="sc">-</span>gmid.df[<span class="dv">1</span>,<span class="dv">1</span>],</span>
<span id="cb65-31"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-31" tabindex="-1"></a>                                              <span class="fu">length</span>(allgenes)<span class="sc">-</span>gmid.df[<span class="dv">1</span>,<span class="dv">3</span>]),</span>
<span id="cb65-32"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-32" tabindex="-1"></a>                         <span class="at">row.names=</span><span class="fu">c</span>(pheno1.label,pheno2.label)),</span>
<span id="cb65-33"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-33" tabindex="-1"></a>                         <span class="at">correct=</span>F)<span class="sc">$</span>p.value</span>
<span id="cb65-34"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-34" tabindex="-1"></a>  </span>
<span id="cb65-35"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-35" tabindex="-1"></a>  LineageA.test <span class="ot">&lt;-</span> <span class="fu">chisq.test</span>(<span class="fu">data.frame</span>(<span class="at">Success=</span><span class="fu">c</span>(gmid.df[<span class="dv">2</span>,<span class="dv">1</span>],gmid.df[<span class="dv">2</span>,<span class="dv">3</span>]),</span>
<span id="cb65-36"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-36" tabindex="-1"></a>                                         <span class="at">Failure=</span><span class="fu">c</span>(<span class="fu">length</span>(allgenes)<span class="sc">-</span>gmid.df[<span class="dv">2</span>,<span class="dv">1</span>],</span>
<span id="cb65-37"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-37" tabindex="-1"></a>                                                   <span class="fu">length</span>(allgenes)<span class="sc">-</span>gmid.df[<span class="dv">2</span>,<span class="dv">3</span>]),                                           <span class="at">row.names=</span><span class="fu">c</span>(pheno1.label,pheno2.label)),</span>
<span id="cb65-38"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-38" tabindex="-1"></a>                              <span class="at">correct=</span>F,<span class="at">simulate.p.value =</span> <span class="cn">TRUE</span>)<span class="sc">$</span>p.value</span>
<span id="cb65-39"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-39" tabindex="-1"></a>  </span>
<span id="cb65-40"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-40" tabindex="-1"></a>  LineageB.test <span class="ot">&lt;-</span> <span class="fu">chisq.test</span>(<span class="fu">data.frame</span>(<span class="at">Success=</span><span class="fu">c</span>(gmid.df[<span class="dv">3</span>,<span class="dv">1</span>],gmid.df[<span class="dv">3</span>,<span class="dv">3</span>]),</span>
<span id="cb65-41"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-41" tabindex="-1"></a>                                         <span class="at">Failure=</span><span class="fu">c</span>(<span class="fu">length</span>(allgenes)<span class="sc">-</span>gmid.df[<span class="dv">3</span>,<span class="dv">1</span>],</span>
<span id="cb65-42"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-42" tabindex="-1"></a>                                                   <span class="fu">length</span>(allgenes)<span class="sc">-</span>gmid.df[<span class="dv">3</span>,<span class="dv">3</span>]),</span>
<span id="cb65-43"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-43" tabindex="-1"></a>                              <span class="at">row.names=</span><span class="fu">c</span>(pheno1.label,pheno2.label)),</span>
<span id="cb65-44"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-44" tabindex="-1"></a>                              <span class="at">correct=</span>F,<span class="at">simulate.p.value =</span> <span class="cn">TRUE</span>)<span class="sc">$</span>p.value</span>
<span id="cb65-45"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-45" tabindex="-1"></a>  </span>
<span id="cb65-46"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-46" tabindex="-1"></a>  pat.test <span class="ot">&lt;-</span> <span class="fu">chisq.test</span>(<span class="fu">data.frame</span>(<span class="at">Success=</span><span class="fu">c</span>(gmid.df[<span class="dv">4</span>,<span class="dv">1</span>],gmid.df[<span class="dv">4</span>,<span class="dv">3</span>]),</span>
<span id="cb65-47"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-47" tabindex="-1"></a>                                    <span class="at">Failure=</span><span class="fu">c</span>(<span class="fu">length</span>(allgenes)<span class="sc">-</span>gmid.df[<span class="dv">4</span>,<span class="dv">1</span>],</span>
<span id="cb65-48"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-48" tabindex="-1"></a>                                              <span class="fu">length</span>(allgenes)<span class="sc">-</span>gmid.df[<span class="dv">4</span>,<span class="dv">3</span>]),</span>
<span id="cb65-49"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-49" tabindex="-1"></a>                         <span class="at">row.names=</span><span class="fu">c</span>(pheno1.label,pheno2.label)),</span>
<span id="cb65-50"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-50" tabindex="-1"></a>                         <span class="at">correct=</span>F)<span class="sc">$</span>p.value</span>
<span id="cb65-51"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-51" tabindex="-1"></a>  </span>
<span id="cb65-52"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-52" tabindex="-1"></a>  <span class="co"># Build table for plotting</span></span>
<span id="cb65-53"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-53" tabindex="-1"></a>  <span class="fu">names</span>(gmid.df) <span class="ot">&lt;-</span> <span class="fu">c</span>(pheno1.label,<span class="st">&quot;Bias&quot;</span>,pheno2.label)</span>
<span id="cb65-54"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-54" tabindex="-1"></a>  gmid.df<span class="sc">$</span>p <span class="ot">&lt;-</span> <span class="fu">c</span>(mat.test,LineageA.test,LineageB.test,pat.test)</span>
<span id="cb65-55"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-55" tabindex="-1"></a>  gmid.df <span class="ot">&lt;-</span> gmid.df[,<span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>)]</span>
<span id="cb65-56"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-56" tabindex="-1"></a>  nsrows <span class="ot">&lt;-</span> <span class="fu">row.names</span>(gmid.df[gmid.df<span class="sc">$</span>p<span class="sc">&gt;</span><span class="fl">0.05</span>,])</span>
<span id="cb65-57"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-57" tabindex="-1"></a>  gmid.df<span class="sc">$</span>p <span class="ot">&lt;-</span> <span class="fu">formatC</span>(gmid.df<span class="sc">$</span>p, <span class="at">format =</span> <span class="st">&quot;e&quot;</span>, <span class="at">digits =</span> <span class="dv">2</span>)</span>
<span id="cb65-58"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-58" tabindex="-1"></a>  gmid.df[nsrows,<span class="st">&quot;p&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;(ns)&quot;</span></span>
<span id="cb65-59"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-59" tabindex="-1"></a>  gmid.df <span class="ot">&lt;-</span> gmid.df[,<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">1</span>)]</span>
<span id="cb65-60"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-60" tabindex="-1"></a>  </span>
<span id="cb65-61"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-61" tabindex="-1"></a>  <span class="co"># Output</span></span>
<span id="cb65-62"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-62" tabindex="-1"></a>  <span class="fu">return</span>(gmid.df)</span>
<span id="cb65-63"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb65-63" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb66-1" tabindex="-1"></a><span class="co"># Execute</span></span>
<span id="cb66-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb66-2" tabindex="-1"></a>chisq.df <span class="ot">&lt;-</span> <span class="fu">AST.chisq</span>(<span class="st">&quot;WL&quot;</span>,WL.plot,<span class="st">&quot;QL&quot;</span>,QL.plot,allgenes)</span></code></pre></div>
<p>We can see from this table that the QL group has more genes with paternal allele-biased transcription than the WL group, but the other categories of allelic bias are comparable between groups:</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:right;">
WL
</th>
<th style="text-align:left;">
Bias
</th>
<th style="text-align:right;">
QL
</th>
<th style="text-align:left;">
p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
20
</td>
<td style="text-align:left;">
mat
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:left;">
(ns)
</td>
</tr>
<tr>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
Lineage A
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
(ns)
</td>
</tr>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
Lineage B
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
(ns)
</td>
</tr>
<tr>
<td style="text-align:right;">
51
</td>
<td style="text-align:left;">
pat
</td>
<td style="text-align:right;">
92
</td>
<td style="text-align:left;">
5.84e-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="data-visualization" class="section level2 hasAnchor" number="3.7">
<h2><span class="header-section-number">3.7</span> Data visualization<a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#data-visualization" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In this final section, the SNP-level counts are aggregated by gene to calculate and plot allelic transcription on <em>p1</em> and <em>p2</em>. Data are plotted for each phenotype, separately, and then combined into a single plot separated by the Chi-squared test result table generated in the previous section.</p>
<p><strong>First</strong>, reorder the data generated by <code>AST.Analysis</code> so that unbiased genes appear first:</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb67-1" tabindex="-1"></a>WL.plot <span class="ot">&lt;-</span> <span class="fu">rbind</span>(WL.plot[WL.plot<span class="sc">$</span>bias.plot<span class="sc">%in%</span><span class="fu">c</span>(<span class="st">&quot;NA&quot;</span>),],</span>
<span id="cb67-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb67-2" tabindex="-1"></a>                           WL.plot[WL.plot<span class="sc">$</span>bias.plot<span class="sc">%in%</span><span class="fu">c</span>(</span>
<span id="cb67-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb67-3" tabindex="-1"></a>                             <span class="st">&quot;mat&quot;</span>, <span class="st">&quot;Lineage A&quot;</span>, <span class="st">&quot;Lineage B&quot;</span>, <span class="st">&quot;pat&quot;</span>),])</span>
<span id="cb67-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb67-4" tabindex="-1"></a>WL.plot<span class="sc">$</span>bias.plot <span class="ot">&lt;-</span> <span class="fu">factor</span>(WL.plot<span class="sc">$</span>bias.plot,</span>
<span id="cb67-5"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb67-5" tabindex="-1"></a>                                      <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;NA&quot;</span>,<span class="st">&quot;mat&quot;</span>, <span class="st">&quot;Lineage A&quot;</span>, </span>
<span id="cb67-6"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb67-6" tabindex="-1"></a>                                                 <span class="st">&quot;Lineage B&quot;</span>, <span class="st">&quot;pat&quot;</span>))</span>
<span id="cb67-7"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb67-7" tabindex="-1"></a></span>
<span id="cb67-8"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb67-8" tabindex="-1"></a>QL.plot <span class="ot">&lt;-</span> <span class="fu">rbind</span>(QL.plot[QL.plot<span class="sc">$</span>bias.plot<span class="sc">%in%</span><span class="fu">c</span>(<span class="st">&quot;NA&quot;</span>),],</span>
<span id="cb67-9"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb67-9" tabindex="-1"></a>                         QL.plot[QL.plot<span class="sc">$</span>bias.plot<span class="sc">%in%</span><span class="fu">c</span>(</span>
<span id="cb67-10"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb67-10" tabindex="-1"></a>                           <span class="st">&quot;mat&quot;</span>, <span class="st">&quot;Lineage A&quot;</span>, <span class="st">&quot;Lineage B&quot;</span>, <span class="st">&quot;pat&quot;</span>),])</span>
<span id="cb67-11"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb67-11" tabindex="-1"></a>QL.plot<span class="sc">$</span>bias.plot <span class="ot">&lt;-</span> <span class="fu">factor</span>(QL.plot<span class="sc">$</span>bias.plot,</span>
<span id="cb67-12"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb67-12" tabindex="-1"></a>                                    <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;NA&quot;</span>,<span class="st">&quot;mat&quot;</span>, <span class="st">&quot;Lineage A&quot;</span>, </span>
<span id="cb67-13"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb67-13" tabindex="-1"></a>                                               <span class="st">&quot;Lineage B&quot;</span>, <span class="st">&quot;pat&quot;</span>))</span></code></pre></div>
<div id="function-ast.collapse" class="section level5 unnumbered hasAnchor">
<h5>Function: <code>AST.collapse</code><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#function-ast.collapse" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p><strong>Next</strong>, use this function to collapse the data generated by <code>AST.Analysis</code> and calculate <em>p1</em> and <em>p2</em> by gene:</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-1" tabindex="-1"></a>AST.collapse <span class="ot">&lt;-</span> <span class="cf">function</span>(data.plot,data.counts,phenotype){</span>
<span id="cb68-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-2" tabindex="-1"></a>  <span class="co"># data.plot = object from AST.Analysis</span></span>
<span id="cb68-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-3" tabindex="-1"></a>  <span class="co"># data.counts = phenotype-specific counts matrix</span></span>
<span id="cb68-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-4" tabindex="-1"></a>  <span class="co"># phenotype = corresponding phenotype of data.plot and data.counts</span></span>
<span id="cb68-5"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-5" tabindex="-1"></a>  </span>
<span id="cb68-6"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-6" tabindex="-1"></a>  <span class="co"># Wrangle data</span></span>
<span id="cb68-7"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-7" tabindex="-1"></a>  data.counts <span class="ot">&lt;-</span> data.counts[,<span class="fu">names</span>(data.counts)<span class="sc">%in%</span>metadata<span class="sc">$</span>sample.id]</span>
<span id="cb68-8"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-8" tabindex="-1"></a>  data.counts<span class="sc">$</span>SNP_gene <span class="ot">&lt;-</span> <span class="fu">row.names</span>(data.counts)</span>
<span id="cb68-9"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-9" tabindex="-1"></a>  data <span class="ot">&lt;-</span> data.plot <span class="sc">%&gt;%</span> </span>
<span id="cb68-10"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-10" tabindex="-1"></a>    <span class="fu">left_join</span>(data.counts, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&#39;SNP_gene&#39;</span> <span class="ot">=</span> <span class="st">&#39;SNP_gene&#39;</span>)) </span>
<span id="cb68-11"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-11" tabindex="-1"></a>  genelist <span class="ot">&lt;-</span> <span class="fu">unique</span>(data<span class="sc">$</span>gene)</span>
<span id="cb68-12"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-12" tabindex="-1"></a>  p1.mean <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb68-13"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-13" tabindex="-1"></a>  p2.mean <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb68-14"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-14" tabindex="-1"></a>  biaslist <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb68-15"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-15" tabindex="-1"></a>  altbias <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb68-16"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-16" tabindex="-1"></a>  </span>
<span id="cb68-17"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-17" tabindex="-1"></a>  <span class="co"># Collapse SNPs by gene</span></span>
<span id="cb68-18"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-18" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(genelist)){</span>
<span id="cb68-19"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-19" tabindex="-1"></a>    tmp <span class="ot">&lt;-</span> data[data<span class="sc">$</span>gene<span class="sc">==</span>genelist[i],]</span>
<span id="cb68-20"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-20" tabindex="-1"></a>    altbias[i] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(tmp<span class="sc">$</span>bias)[<span class="dv">1</span>]</span>
<span id="cb68-21"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-21" tabindex="-1"></a>    <span class="cf">if</span>(<span class="sc">!</span><span class="fu">any</span>(tmp<span class="sc">$</span>bias<span class="sc">==</span><span class="st">&quot;NA&quot;</span>) <span class="sc">&amp;</span> <span class="fu">length</span>(tmp[<span class="sc">!</span>tmp<span class="sc">$</span>bias.plot<span class="sc">==</span><span class="st">&quot;NA&quot;</span>,<span class="st">&quot;p1&quot;</span>])<span class="sc">&gt;</span><span class="dv">0</span>){</span>
<span id="cb68-22"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-22" tabindex="-1"></a>      tmp.sub <span class="ot">&lt;-</span> tmp[<span class="sc">!</span>tmp<span class="sc">$</span>bias.plot<span class="sc">==</span><span class="st">&quot;NA&quot;</span>,]</span>
<span id="cb68-23"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-23" tabindex="-1"></a>      <span class="co"># Split count matrices by Lineage and parent of origin for plotting</span></span>
<span id="cb68-24"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-24" tabindex="-1"></a>      p1.pat <span class="ot">&lt;-</span> tmp.sub[,metadata[metadata<span class="sc">$</span>parent<span class="sc">%in%</span><span class="fu">c</span>(<span class="st">&quot;M&quot;</span>)<span class="sc">&amp;</span>metadata<span class="sc">$</span>lineage<span class="sc">==</span><span class="st">&quot;B&quot;</span><span class="sc">&amp;</span>metadata<span class="sc">$</span>phenotype<span class="sc">==</span>phenotype,<span class="st">&quot;sample.id&quot;</span>]]</span>
<span id="cb68-25"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-25" tabindex="-1"></a>      p1.mat <span class="ot">&lt;-</span> tmp.sub[,metadata[metadata<span class="sc">$</span>parent<span class="sc">%in%</span><span class="fu">c</span>(<span class="st">&quot;F&quot;</span>)<span class="sc">&amp;</span>metadata<span class="sc">$</span>lineage<span class="sc">==</span><span class="st">&quot;B&quot;</span><span class="sc">&amp;</span>metadata<span class="sc">$</span>phenotype<span class="sc">==</span>phenotype,<span class="st">&quot;sample.id&quot;</span>]]</span>
<span id="cb68-26"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-26" tabindex="-1"></a>      p2.pat <span class="ot">&lt;-</span> tmp.sub[,metadata[metadata<span class="sc">$</span>parent<span class="sc">%in%</span><span class="fu">c</span>(<span class="st">&quot;M&quot;</span>)<span class="sc">&amp;</span>metadata<span class="sc">$</span>lineage<span class="sc">==</span><span class="st">&quot;A&quot;</span><span class="sc">&amp;</span>metadata<span class="sc">$</span>phenotype<span class="sc">==</span>phenotype,<span class="st">&quot;sample.id&quot;</span>]]</span>
<span id="cb68-27"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-27" tabindex="-1"></a>      p2.mat <span class="ot">&lt;-</span> tmp.sub[,metadata[metadata<span class="sc">$</span>parent<span class="sc">%in%</span><span class="fu">c</span>(<span class="st">&quot;F&quot;</span>)<span class="sc">&amp;</span>metadata<span class="sc">$</span>lineage<span class="sc">==</span><span class="st">&quot;A&quot;</span><span class="sc">&amp;</span>metadata<span class="sc">$</span>phenotype<span class="sc">==</span>phenotype,<span class="st">&quot;sample.id&quot;</span>]]</span>
<span id="cb68-28"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-28" tabindex="-1"></a>      p1.mean.x <span class="ot">&lt;-</span> <span class="fu">mean</span>(<span class="fu">sum</span>(p1.pat)<span class="sc">/</span>(<span class="fu">sum</span>(p1.mat)<span class="sc">+</span><span class="fu">sum</span>(p1.pat)))</span>
<span id="cb68-29"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-29" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">is.nan</span>(p1.mean.x)){p1.mean.x <span class="ot">&lt;-</span> <span class="dv">0</span>}</span>
<span id="cb68-30"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-30" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">is.infinite</span>(p1.mean.x)){p1.mean.x <span class="ot">&lt;-</span> <span class="dv">1</span>}</span>
<span id="cb68-31"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-31" tabindex="-1"></a>      p1.mean[i] <span class="ot">&lt;-</span> p1.mean.x</span>
<span id="cb68-32"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-32" tabindex="-1"></a>      p2.mean.x <span class="ot">&lt;-</span> <span class="fu">mean</span>(<span class="fu">sum</span>(p2.mat)<span class="sc">/</span>(<span class="fu">sum</span>(p2.mat)<span class="sc">+</span><span class="fu">sum</span>(p2.pat)))</span>
<span id="cb68-33"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-33" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">is.nan</span>(p2.mean.x)){p2.mean.x <span class="ot">&lt;-</span> <span class="dv">0</span>}</span>
<span id="cb68-34"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-34" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">is.infinite</span>(p2.mean.x)){p2.mean.x <span class="ot">&lt;-</span> <span class="dv">1</span>}</span>
<span id="cb68-35"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-35" tabindex="-1"></a>      p2.mean[i] <span class="ot">&lt;-</span> p2.mean.x</span>
<span id="cb68-36"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-36" tabindex="-1"></a>      biaslist[i] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(tmp.sub<span class="sc">$</span>bias.plot[<span class="dv">1</span>])</span>
<span id="cb68-37"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-37" tabindex="-1"></a>    }<span class="cf">else</span>{</span>
<span id="cb68-38"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-38" tabindex="-1"></a>      p1.mean[i] <span class="ot">&lt;-</span> <span class="fu">mean</span>(tmp<span class="sc">$</span>p1)</span>
<span id="cb68-39"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-39" tabindex="-1"></a>      p2.mean[i] <span class="ot">&lt;-</span> <span class="fu">mean</span>(tmp<span class="sc">$</span>p2)</span>
<span id="cb68-40"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-40" tabindex="-1"></a>      biaslist[i] <span class="ot">&lt;-</span> <span class="st">&quot;NA&quot;</span>}}</span>
<span id="cb68-41"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-41" tabindex="-1"></a>  return.data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene=</span><span class="fu">unlist</span>(genelist),</span>
<span id="cb68-42"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-42" tabindex="-1"></a>                            <span class="at">bias.plot=</span><span class="fu">unlist</span>(biaslist),</span>
<span id="cb68-43"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-43" tabindex="-1"></a>                            <span class="at">p1=</span><span class="fu">unlist</span>(p1.mean),</span>
<span id="cb68-44"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-44" tabindex="-1"></a>                            <span class="at">p2=</span><span class="fu">unlist</span>(p2.mean),</span>
<span id="cb68-45"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-45" tabindex="-1"></a>                            <span class="at">altbias=</span><span class="fu">unlist</span>(altbias))</span>
<span id="cb68-46"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-46" tabindex="-1"></a>  return.data<span class="sc">$</span>bias.plot <span class="ot">&lt;-</span> <span class="fu">factor</span>(return.data<span class="sc">$</span>bias.plot,</span>
<span id="cb68-47"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-47" tabindex="-1"></a>                                  <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;NA&quot;</span>,<span class="st">&quot;mat&quot;</span>, <span class="st">&quot;Lineage A&quot;</span>, <span class="st">&quot;Lineage B&quot;</span>, <span class="st">&quot;pat&quot;</span>))</span>
<span id="cb68-48"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-48" tabindex="-1"></a>  return.data <span class="ot">&lt;-</span> return.data[<span class="fu">order</span>(return.data<span class="sc">$</span>bias.plot),]</span>
<span id="cb68-49"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-49" tabindex="-1"></a>  </span>
<span id="cb68-50"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-50" tabindex="-1"></a>  <span class="co"># Output</span></span>
<span id="cb68-51"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-51" tabindex="-1"></a>  <span class="fu">return</span>(return.data)</span>
<span id="cb68-52"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb68-52" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb69-1" tabindex="-1"></a><span class="co"># Execute</span></span>
<span id="cb69-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb69-2" tabindex="-1"></a>WL.collapse <span class="ot">&lt;-</span> <span class="fu">AST.collapse</span>(WL.plot,WL_counts_normalized,<span class="st">&quot;WL&quot;</span>)</span>
<span id="cb69-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb69-3" tabindex="-1"></a>QL.collapse <span class="ot">&lt;-</span> <span class="fu">AST.collapse</span>(QL.plot,QL_counts_normalized,<span class="st">&quot;QL&quot;</span>)</span>
<span id="cb69-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb69-4" tabindex="-1"></a></span>
<span id="cb69-5"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb69-5" tabindex="-1"></a><span class="co"># Save output</span></span>
<span id="cb69-6"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb69-6" tabindex="-1"></a><span class="fu">write.csv</span>(WL.collapse,<span class="st">&quot;AST-tutorial/ANALYSIS/RESULTS/WLAST_collapse.csv&quot;</span>,<span class="at">row.names=</span>F)</span>
<span id="cb69-7"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb69-7" tabindex="-1"></a><span class="fu">write.csv</span>(QL.collapse,<span class="st">&quot;AST-tutorial/ANALYSIS/RESULTS/QLAST_collapse.csv&quot;</span>,<span class="at">row.names=</span>F)</span></code></pre></div>
<p><strong>Finally</strong>, generate scatter plots of allele-specific transcription joined by the table of Chi-squared test results.</p>
</div>
<div id="function-ast.scatter" class="section level5 unnumbered hasAnchor">
<h5>Function: <code>AST.scatter</code><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#function-ast.scatter" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>This function generates a scatter plot of each gene by <em>p1</em> and <em>p2</em> for each phenotype. The points in these plots are assigned a color to represent its expression status: black is maternal, green is lineage A, gold is lineage B, blue is paternal, and grey is not significant. For consistency across studies, the colors for maternal, paternal, and unbiased genes are fixed, but the colors for each lineage can be customized. To help with visualizing the local density of points, each point is assigned its color value on a gradient between darker and lighter tones, indicating higher and lower local densities, respectively.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-1" tabindex="-1"></a>AST.scatter <span class="ot">&lt;-</span> <span class="cf">function</span>(data,title,</span>
<span id="cb70-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-2" tabindex="-1"></a>                         <span class="at">LineageA.color.dark=</span><span class="st">&quot;#058762&quot;</span>,<span class="at">LineageA.color.light=</span><span class="st">&quot;#4dc4a2&quot;</span>,</span>
<span id="cb70-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-3" tabindex="-1"></a>                         <span class="at">LineageB.color.dark=</span><span class="st">&quot;#f7af05&quot;</span>,<span class="at">LineageB.color.light=</span><span class="st">&quot;#f7ca60&quot;</span>){</span>
<span id="cb70-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-4" tabindex="-1"></a>  <span class="co"># data = object from AST.collapse</span></span>
<span id="cb70-5"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-5" tabindex="-1"></a>  <span class="co"># title = string, title of plot</span></span>
<span id="cb70-6"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-6" tabindex="-1"></a>  </span>
<span id="cb70-7"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-7" tabindex="-1"></a>  get_density <span class="ot">&lt;-</span> <span class="cf">function</span>(x, y, ...){</span>
<span id="cb70-8"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-8" tabindex="-1"></a>    dens <span class="ot">&lt;-</span> MASS<span class="sc">::</span><span class="fu">kde2d</span>(x, y, ...)</span>
<span id="cb70-9"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-9" tabindex="-1"></a>    ix <span class="ot">&lt;-</span> <span class="fu">findInterval</span>(x, dens<span class="sc">$</span>x)</span>
<span id="cb70-10"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-10" tabindex="-1"></a>    iy <span class="ot">&lt;-</span> <span class="fu">findInterval</span>(y, dens<span class="sc">$</span>y)</span>
<span id="cb70-11"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-11" tabindex="-1"></a>    ii <span class="ot">&lt;-</span> <span class="fu">cbind</span>(ix, iy)</span>
<span id="cb70-12"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-12" tabindex="-1"></a>    <span class="fu">return</span>(dens<span class="sc">$</span>z[ii])}</span>
<span id="cb70-13"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-13" tabindex="-1"></a>  </span>
<span id="cb70-14"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-14" tabindex="-1"></a>  data<span class="sc">$</span>bias.plot <span class="ot">&lt;-</span> <span class="fu">factor</span>(data<span class="sc">$</span>bias.plot)</span>
<span id="cb70-15"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-15" tabindex="-1"></a>  biases <span class="ot">&lt;-</span> <span class="fu">levels</span>(data<span class="sc">$</span>bias.plot)</span>
<span id="cb70-16"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-16" tabindex="-1"></a>  data<span class="sc">$</span>color <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb70-17"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-17" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(biases)){</span>
<span id="cb70-18"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-18" tabindex="-1"></a>    data.sub <span class="ot">&lt;-</span> data[data<span class="sc">$</span>bias.plot<span class="sc">==</span>biases[i],]</span>
<span id="cb70-19"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-19" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(data.sub<span class="sc">$</span>gene)<span class="sc">&gt;</span><span class="dv">2</span>){</span>
<span id="cb70-20"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-20" tabindex="-1"></a>      density <span class="ot">&lt;-</span> <span class="fu">get_density</span>(data.sub<span class="sc">$</span>p1, data.sub<span class="sc">$</span>p2, <span class="at">n =</span> <span class="dv">100</span>)</span>
<span id="cb70-21"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-21" tabindex="-1"></a>      <span class="cf">if</span>(biases[i]<span class="sc">==</span><span class="st">&quot;NA&quot;</span>){pal <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="at">colors =</span> <span class="fu">c</span>(<span class="st">&quot;grey90&quot;</span>, <span class="st">&quot;grey70&quot;</span>))(<span class="dv">60</span>)}</span>
<span id="cb70-22"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-22" tabindex="-1"></a>      <span class="cf">if</span>(biases[i]<span class="sc">==</span><span class="st">&quot;pat&quot;</span>){pal <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="at">colors =</span> <span class="fu">c</span>(<span class="st">&quot;#a1d2ed&quot;</span>, <span class="st">&quot;#02a0f5&quot;</span>))(<span class="dv">60</span>)}</span>
<span id="cb70-23"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-23" tabindex="-1"></a>      <span class="cf">if</span>(biases[i]<span class="sc">==</span><span class="st">&quot;mat&quot;</span>){pal <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="at">colors =</span> <span class="fu">c</span>(<span class="st">&quot;grey50&quot;</span>, <span class="st">&quot;black&quot;</span>))(<span class="dv">60</span>)}</span>
<span id="cb70-24"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-24" tabindex="-1"></a>      <span class="cf">if</span>(biases[i]<span class="sc">==</span><span class="st">&quot;Lineage A&quot;</span>){pal <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="at">colors =</span> <span class="fu">c</span>(LineageA.color.light,</span>
<span id="cb70-25"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-25" tabindex="-1"></a>                                                                    LineageA.color.dark))(<span class="dv">60</span>)}</span>
<span id="cb70-26"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-26" tabindex="-1"></a>      <span class="cf">if</span>(biases[i]<span class="sc">==</span><span class="st">&quot;Lineage B&quot;</span>){pal <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="at">colors =</span> <span class="fu">c</span>(LineageB.color.light,</span>
<span id="cb70-27"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-27" tabindex="-1"></a>                                                                    LineageB.color.dark))(<span class="dv">60</span>)}</span>
<span id="cb70-28"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-28" tabindex="-1"></a>      data[data<span class="sc">$</span>bias.plot<span class="sc">==</span>biases[i],<span class="st">&quot;color&quot;</span>] <span class="ot">&lt;-</span> <span class="fu">smoothPalette</span>(density,<span class="at">pal=</span>pal)</span>
<span id="cb70-29"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-29" tabindex="-1"></a>    }</span>
<span id="cb70-30"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-30" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(data.sub<span class="sc">$</span>gene)<span class="sc">&lt;=</span><span class="dv">2</span>){</span>
<span id="cb70-31"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-31" tabindex="-1"></a>      <span class="cf">if</span>(biases[i]<span class="sc">==</span><span class="st">&quot;NA&quot;</span>){pal <span class="ot">&lt;-</span> <span class="st">&quot;grey90&quot;</span>}</span>
<span id="cb70-32"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-32" tabindex="-1"></a>      <span class="cf">if</span>(biases[i]<span class="sc">==</span><span class="st">&quot;pat&quot;</span>){pal <span class="ot">&lt;-</span> <span class="st">&quot;#a1d2ed&quot;</span>}</span>
<span id="cb70-33"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-33" tabindex="-1"></a>      <span class="cf">if</span>(biases[i]<span class="sc">==</span><span class="st">&quot;mat&quot;</span>){pal <span class="ot">&lt;-</span> <span class="st">&quot;grey50&quot;</span>}</span>
<span id="cb70-34"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-34" tabindex="-1"></a>      <span class="cf">if</span>(biases[i]<span class="sc">==</span><span class="st">&quot;Lineage A&quot;</span>){pal <span class="ot">&lt;-</span> LineageA.color.light}</span>
<span id="cb70-35"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-35" tabindex="-1"></a>      <span class="cf">if</span>(biases[i]<span class="sc">==</span><span class="st">&quot;Lineage B&quot;</span>){pal <span class="ot">&lt;-</span> LineageB.color.light}</span>
<span id="cb70-36"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-36" tabindex="-1"></a>      data[data<span class="sc">$</span>bias.plot<span class="sc">==</span>biases[i],<span class="st">&quot;color&quot;</span>] <span class="ot">&lt;-</span> pal</span>
<span id="cb70-37"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-37" tabindex="-1"></a>    }</span>
<span id="cb70-38"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-38" tabindex="-1"></a>  }</span>
<span id="cb70-39"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-39" tabindex="-1"></a>  data <span class="ot">&lt;-</span> data[,<span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">6</span>)]</span>
<span id="cb70-40"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-40" tabindex="-1"></a>  data <span class="ot">&lt;-</span> data[<span class="sc">!</span><span class="fu">duplicated</span>(data),]</span>
<span id="cb70-41"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-41" tabindex="-1"></a>  </span>
<span id="cb70-42"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-42" tabindex="-1"></a>  <span class="co"># Generate plot</span></span>
<span id="cb70-43"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-43" tabindex="-1"></a>  g <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x=</span>p1, <span class="at">y=</span>p2,</span>
<span id="cb70-44"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-44" tabindex="-1"></a>                        <span class="at">color=</span>color)) <span class="sc">+</span></span>
<span id="cb70-45"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-45" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">3</span>) <span class="sc">+</span> <span class="fu">theme_prism</span>() <span class="sc">+</span></span>
<span id="cb70-46"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-46" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="fu">expression</span>(<span class="fu">bold</span>(<span class="fu">paste</span>(<span class="st">&quot;% A allele in &quot;</span>,B[mother],</span>
<span id="cb70-47"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-47" tabindex="-1"></a>                               <span class="st">&quot; x &quot;</span>,A[father],<span class="at">sep=</span><span class="st">&quot;&quot;</span>)))) <span class="sc">+</span></span>
<span id="cb70-48"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-48" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="fu">expression</span>(<span class="fu">bold</span>(<span class="fu">paste</span>(<span class="st">&quot;% A allele in &quot;</span>,A[mother],</span>
<span id="cb70-49"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-49" tabindex="-1"></a>                               <span class="st">&quot; x &quot;</span>,B[father],<span class="at">sep=</span><span class="st">&quot;&quot;</span>)))) <span class="sc">+</span></span>
<span id="cb70-50"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-50" tabindex="-1"></a>    <span class="fu">ggtitle</span>(title) <span class="sc">+</span></span>
<span id="cb70-51"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-51" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">18</span>),</span>
<span id="cb70-52"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-52" tabindex="-1"></a>          <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>)) <span class="sc">+</span></span>
<span id="cb70-53"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-53" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">alpha=</span>F, <span class="at">color=</span>F) <span class="sc">+</span></span>
<span id="cb70-54"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-54" tabindex="-1"></a>    <span class="fu">scale_x_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, .<span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb70-55"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-55" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, .<span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb70-56"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-56" tabindex="-1"></a>    <span class="fu">scale_colour_identity</span>()</span>
<span id="cb70-57"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-57" tabindex="-1"></a>  </span>
<span id="cb70-58"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-58" tabindex="-1"></a>  <span class="co"># Output</span></span>
<span id="cb70-59"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-59" tabindex="-1"></a>  <span class="fu">return</span>(g)</span>
<span id="cb70-60"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb70-60" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb71-1" tabindex="-1"></a><span class="co"># Execute for WL</span></span>
<span id="cb71-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb71-2" tabindex="-1"></a>WL.scatter <span class="ot">&lt;-</span> <span class="fu">AST.scatter</span>(WL.collapse,<span class="st">&quot;192hpf WL transcription&quot;</span>,</span>
<span id="cb71-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb71-3" tabindex="-1"></a>                          <span class="at">LineageA.color.dark=</span><span class="st">&quot;#427447&quot;</span>,<span class="at">LineageA.color.light=</span><span class="st">&quot;#C1C9B5&quot;</span>,</span>
<span id="cb71-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb71-4" tabindex="-1"></a>                          <span class="at">LineageB.color.dark=</span><span class="st">&quot;#d09d6f&quot;</span>,<span class="at">LineageB.color.light=</span><span class="st">&quot;#f7dcc3&quot;</span>)</span>
<span id="cb71-5"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb71-5" tabindex="-1"></a>WL.scatter</span></code></pre></div>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-46-1.png" width="672" /></p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb72-1" tabindex="-1"></a><span class="co"># Execute for QL</span></span>
<span id="cb72-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb72-2" tabindex="-1"></a>QL.scatter <span class="ot">&lt;-</span> <span class="fu">AST.scatter</span>(QL.collapse,<span class="st">&quot;192hpf QL transcription&quot;</span>,</span>
<span id="cb72-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb72-3" tabindex="-1"></a>                          <span class="at">LineageA.color.dark=</span><span class="st">&quot;#427447&quot;</span>,<span class="at">LineageA.color.light=</span><span class="st">&quot;#C1C9B5&quot;</span>,</span>
<span id="cb72-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb72-4" tabindex="-1"></a>                          <span class="at">LineageB.color.dark=</span><span class="st">&quot;#d09d6f&quot;</span>,<span class="at">LineageB.color.light=</span><span class="st">&quot;#f7dcc3&quot;</span>)</span>
<span id="cb72-5"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb72-5" tabindex="-1"></a>QL.scatter</span></code></pre></div>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-47-1.png" width="672" /></p>
</div>
<div id="function-ast.scatter.table" class="section level5 unnumbered hasAnchor">
<h5>Function: <code>AST.scatter.table</code><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#function-ast.scatter.table" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>Plot the Chi-squared results table generated by <code>AST.chisq</code>.</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-1" tabindex="-1"></a>AST.scatter.table <span class="ot">&lt;-</span> <span class="cf">function</span>(chisq.df,</span>
<span id="cb73-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-2" tabindex="-1"></a>                              <span class="at">LineageA.color=</span><span class="st">&quot;#009e73&quot;</span>,</span>
<span id="cb73-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-3" tabindex="-1"></a>                              <span class="at">LineageB.color=</span><span class="st">&quot;#e69f00&quot;</span>){</span>
<span id="cb73-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-4" tabindex="-1"></a>  <span class="co"># chisq.table = table generated by AST.chisq</span></span>
<span id="cb73-5"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-5" tabindex="-1"></a>  </span>
<span id="cb73-6"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-6" tabindex="-1"></a>  <span class="co"># Build table plot</span></span>
<span id="cb73-7"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-7" tabindex="-1"></a>  gmid.df <span class="ot">&lt;-</span> chisq.df</span>
<span id="cb73-8"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-8" tabindex="-1"></a>  </span>
<span id="cb73-9"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-9" tabindex="-1"></a>  cols <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="st">&quot;black&quot;</span>, <span class="fu">nrow</span>(gmid.df), <span class="fu">ncol</span>(gmid.df))</span>
<span id="cb73-10"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-10" tabindex="-1"></a>  cols[<span class="dv">1</span>,<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">&quot;#000000&quot;</span></span>
<span id="cb73-11"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-11" tabindex="-1"></a>  cols[<span class="dv">2</span>,<span class="dv">2</span>] <span class="ot">&lt;-</span> LineageA.color</span>
<span id="cb73-12"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-12" tabindex="-1"></a>  cols[<span class="dv">3</span>,<span class="dv">2</span>] <span class="ot">&lt;-</span> LineageB.color</span>
<span id="cb73-13"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-13" tabindex="-1"></a>  cols[<span class="dv">4</span>,<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">&quot;#56b4e9&quot;</span></span>
<span id="cb73-14"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-14" tabindex="-1"></a>          </span>
<span id="cb73-15"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-15" tabindex="-1"></a>  ccols <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="st">&quot;white&quot;</span>, <span class="fu">nrow</span>(gmid.df), <span class="fu">ncol</span>(gmid.df))</span>
<span id="cb73-16"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-16" tabindex="-1"></a>  </span>
<span id="cb73-17"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-17" tabindex="-1"></a>  ccols[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>,<span class="dv">3</span>] <span class="ot">&lt;-</span> <span class="st">&quot;#f4efea&quot;</span></span>
<span id="cb73-18"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-18" tabindex="-1"></a>  ccols[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;#f4efea&quot;</span></span>
<span id="cb73-19"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-19" tabindex="-1"></a>  ccols[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>,<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">&quot;#e4d8d1&quot;</span></span>
<span id="cb73-20"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-20" tabindex="-1"></a>                                </span>
<span id="cb73-21"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-21" tabindex="-1"></a>  cfonts <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="st">&quot;bold&quot;</span>, <span class="fu">nrow</span>(gmid.df), <span class="fu">ncol</span>(gmid.df))</span>
<span id="cb73-22"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-22" tabindex="-1"></a>                              </span>
<span id="cb73-23"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-23" tabindex="-1"></a>  gmid.df[<span class="dv">2</span>,<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">&quot;A&quot;</span></span>
<span id="cb73-24"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-24" tabindex="-1"></a>  gmid.df[<span class="dv">3</span>,<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">&quot;B&quot;</span></span>
<span id="cb73-25"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-25" tabindex="-1"></a>                              </span>
<span id="cb73-26"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-26" tabindex="-1"></a>  tt <span class="ot">&lt;-</span> <span class="fu">ttheme_default</span>(<span class="at">core=</span><span class="fu">list</span>(<span class="at">fg_params =</span> <span class="fu">list</span>(<span class="at">col =</span> cols,</span>
<span id="cb73-27"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-27" tabindex="-1"></a>                                                  <span class="at">cex =</span> <span class="dv">1</span>,</span>
<span id="cb73-28"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-28" tabindex="-1"></a>                                                  <span class="at">fontface =</span> cfonts),</span>
<span id="cb73-29"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-29" tabindex="-1"></a>                                 <span class="at">bg_params =</span> <span class="fu">list</span>(<span class="at">col=</span><span class="cn">NA</span>, <span class="at">fill =</span> ccols),</span>
<span id="cb73-30"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-30" tabindex="-1"></a>                                 <span class="at">padding.h=</span><span class="fu">unit</span>(<span class="dv">2</span>, <span class="st">&quot;mm&quot;</span>)),</span>
<span id="cb73-31"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-31" tabindex="-1"></a>                       <span class="at">rowhead=</span><span class="fu">list</span>(<span class="at">bg_params =</span> <span class="fu">list</span>(<span class="at">col=</span><span class="cn">NA</span>)),</span>
<span id="cb73-32"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-32" tabindex="-1"></a>                       <span class="at">colhead=</span><span class="fu">list</span>(<span class="at">bg_params =</span> <span class="fu">list</span>(<span class="at">fill =</span><span class="fu">c</span>(<span class="st">&quot;#f4efea&quot;</span>,</span>
<span id="cb73-33"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-33" tabindex="-1"></a>                                                             <span class="st">&quot;#e4d8d1&quot;</span>,</span>
<span id="cb73-34"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-34" tabindex="-1"></a>                                                             <span class="st">&quot;#f4efea&quot;</span>,</span>
<span id="cb73-35"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-35" tabindex="-1"></a>                                                             <span class="st">&quot;white&quot;</span>)),</span>
<span id="cb73-36"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-36" tabindex="-1"></a>                                    <span class="at">fg_params =</span> <span class="fu">list</span>(<span class="at">rot=</span><span class="dv">90</span>,</span>
<span id="cb73-37"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-37" tabindex="-1"></a>                                                     <span class="at">cex =</span> <span class="dv">1</span>,<span class="at">col=</span><span class="fu">c</span>(<span class="st">&quot;black&quot;</span>,</span>
<span id="cb73-38"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-38" tabindex="-1"></a>                                                                   <span class="st">&quot;black&quot;</span>,</span>
<span id="cb73-39"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-39" tabindex="-1"></a>                                                                   <span class="st">&quot;black&quot;</span>,</span>
<span id="cb73-40"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-40" tabindex="-1"></a>                                                                   <span class="st">&quot;white&quot;</span>))))</span>
<span id="cb73-41"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-41" tabindex="-1"></a>                              </span>
<span id="cb73-42"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-42" tabindex="-1"></a> <span class="co"># Generate plot</span></span>
<span id="cb73-43"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-43" tabindex="-1"></a> gmid <span class="ot">&lt;-</span> <span class="fu">tableGrob</span>(gmid.df, <span class="at">rows =</span> <span class="cn">NULL</span>, <span class="at">theme=</span>tt)</span>
<span id="cb73-44"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-44" tabindex="-1"></a>                              </span>
<span id="cb73-45"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-45" tabindex="-1"></a> <span class="co"># Output</span></span>
<span id="cb73-46"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-46" tabindex="-1"></a> <span class="fu">return</span>(gmid)</span>
<span id="cb73-47"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb73-47" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb74-1" tabindex="-1"></a><span class="co"># Execute</span></span>
<span id="cb74-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb74-2" tabindex="-1"></a>chisq.table <span class="ot">&lt;-</span> <span class="fu">AST.scatter.table</span>(chisq.df,</span>
<span id="cb74-3"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb74-3" tabindex="-1"></a>                                 <span class="at">LineageA.color=</span><span class="st">&quot;#427447&quot;</span>,</span>
<span id="cb74-4"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb74-4" tabindex="-1"></a>                                 <span class="at">LineageB.color=</span><span class="st">&quot;#d09d6f&quot;</span>)</span>
<span id="cb74-5"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb74-5" tabindex="-1"></a><span class="fu">plot</span>(chisq.table)</span></code></pre></div>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-49-1.png" width="672" /></p>
<p>Combine the plots into a single figure to replicate <strong>Figure 4</strong> from <a href="https://doi.org/10.1101/2024.06.09.598129">Bresnahan <em>et al.</em>, 2024, “Intragenomic conflict underlies extreme phenotypic plasticity in queen-worker caste determination in honey bees (<em>Apis mellifera</em>)”, <em>bioRxiv</em></a>.</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb75-1" tabindex="-1"></a>fig4 <span class="ot">&lt;-</span> <span class="fu">arrangeGrob</span>(WL.scatter, chisq.table, QL.scatter, <span class="at">widths=</span><span class="fu">c</span>(<span class="dv">5</span>,<span class="fl">2.5</span>,<span class="dv">5</span>))</span>
<span id="cb75-2"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb75-2" tabindex="-1"></a>fig4</span></code></pre></div>
<div class="float">
<img src="AST-tutorial/fig4.png" alt="Figure 4. Queen-destined larvae show enriched paternal allele-biased transcription relative to worker-destined larvae. Allele-specific transcriptomes were assessed in F2 worker-destined larvae (WL) and queen-destined larvae (QL) collected from a reciprocal cross between different stocks of European honey bees. The x-axis represents, for each transcript, the proportion of lineage A reads in larvae with a lineage B mother and lineage A father (p1). The y-axis represents, for each transcript, the proportion of lineage A reads in larvae with a lineage A mother and lineage B father (p2). Each color represents a transcript which is significantly biased at all tested SNP positions: black is maternal (mat), green is lineage A, gold is lineage B, blue is paternal (pat), and grey is not significant. Center table: the number of transcripts showing each category of allelic bias and p-values for Chi-squared tests of independence for comparisons between the castes are indicated (NS = not significant). Significance of allele-biased transcription was determined using the overlap between two statistical tests: a general linear mixed model (GLIMMIX), and a Storer-Kim binomial exact test along with thresholds of p1&lt;0.4 and p2&gt;0.6 for maternal bias, p1&gt;0.6 and p2&lt;0.4 for paternal bias, p1&lt;0.4 and p2&lt;0.4 for lineage B bias and p1&gt;0.6 and p2&gt;0.6 for lineage A bias." />
<div class="figcaption"><strong>Figure 4.</strong> Queen-destined larvae show enriched paternal allele-biased transcription relative to worker-destined larvae. Allele-specific transcriptomes were assessed in F2 worker-destined larvae (WL) and queen-destined larvae (QL) collected from a reciprocal cross between different stocks of European honey bees. The x-axis represents, for each transcript, the proportion of lineage A reads in larvae with a lineage B mother and lineage A father (<em>p1</em>). The y-axis represents, for each transcript, the proportion of lineage A reads in larvae with a lineage A mother and lineage B father (<em>p2</em>). Each color represents a transcript which is significantly biased at all tested SNP positions: black is maternal (mat), green is lineage A, gold is lineage B, blue is paternal (pat), and grey is not significant. <strong>Center table</strong>: the number of transcripts showing each category of allelic bias and <em>p</em>-values for Chi-squared tests of independence for comparisons between the castes are indicated (NS = not significant). Significance of allele-biased transcription was determined using the overlap between two statistical tests: a general linear mixed model (GLIMMIX), and a Storer-Kim binomial exact test along with thresholds of <em>p1</em>&lt;0.4 and <em>p2</em>&gt;0.6 for maternal bias, <em>p1</em>&gt;0.6 and <em>p2</em>&lt;0.4 for paternal bias, <em>p1</em>&lt;0.4 and <em>p2</em>&lt;0.4 for lineage B bias and <em>p1</em>&gt;0.6 and <em>p2</em>&gt;0.6 for lineage A bias.</div>
</div>
</div>
</div>
<div id="session-info" class="section level2 hasAnchor" number="3.8">
<h2><span class="header-section-number">3.8</span> Session info<a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#session-info" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="snp-level-parent-specific-read-counts-to-allele-specific-transcription-1.html#cb76-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.3.2 (2023-10-31)
## Platform: aarch64-apple-darwin20 (64-bit)
## Running under: macOS Ventura 13.1
## 
## Matrix products: default
## BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib 
## LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## time zone: America/New_York
## tzcode source: internal
## 
## attached base packages:
##  [1] stats4    grid      parallel  stats     graphics  grDevices utils    
##  [8] datasets  methods   base     
## 
## other attached packages:
##  [1] GenomicFeatures_1.54.4      AnnotationDbi_1.64.1       
##  [3] genomation_1.34.0           DESeq2_1.42.1              
##  [5] SummarizedExperiment_1.32.0 Biobase_2.62.0             
##  [7] MatrixGenerics_1.14.0       matrixStats_1.1.0          
##  [9] GenomicRanges_1.54.1        GenomeInfoDb_1.38.1        
## [11] IRanges_2.36.0              S4Vectors_0.40.2           
## [13] BiocGenerics_0.48.1         ggprism_1.0.4              
## [15] tagcloud_0.6                ggpubr_0.6.0               
## [17] doParallel_1.0.17           iterators_1.0.14           
## [19] foreach_1.5.2               gridExtra_2.3              
## [21] car_3.1-2                   carData_3.0-5              
## [23] lmerTest_3.1-3              lme4_1.1-35.1              
## [25] Matrix_1.6-3                tryCatchLog_1.3.1          
## [27] Rfast_2.1.0                 RcppParallel_5.1.7         
## [29] RcppZiggurat_0.1.6          Rcpp_1.0.11                
## [31] plyr_1.8.9                  lubridate_1.9.3            
## [33] forcats_1.0.0               stringr_1.5.1              
## [35] dplyr_1.1.4                 purrr_1.0.2                
## [37] readr_2.1.4                 tidyr_1.3.0                
## [39] tibble_3.2.1                ggplot2_3.4.4              
## [41] tidyverse_2.0.0             kableExtra_1.3.4           
## 
## loaded via a namespace (and not attached):
##   [1] splines_4.3.2            BiocIO_1.12.0           
##   [3] bitops_1.0-7             filelock_1.0.2          
##   [5] XML_3.99-0.15            lifecycle_1.0.4         
##   [7] rstatix_0.7.2            lattice_0.22-5          
##   [9] MASS_7.3-60              backports_1.4.1         
##  [11] magrittr_2.0.3           sass_0.4.7              
##  [13] rmarkdown_2.25           jquerylib_0.1.4         
##  [15] yaml_2.3.7               plotrix_3.8-4           
##  [17] DBI_1.1.3                minqa_1.2.6             
##  [19] RColorBrewer_1.1-3       abind_1.4-5             
##  [21] zlibbioc_1.48.0          rvest_1.0.3             
##  [23] RCurl_1.98-1.13          rappdirs_0.3.3          
##  [25] GenomeInfoDbData_1.2.11  svglite_2.1.2           
##  [27] codetools_0.2-19         DelayedArray_0.28.0     
##  [29] xml2_1.3.5               tidyselect_1.2.0        
##  [31] futile.logger_1.4.3      farver_2.1.1            
##  [33] BiocFileCache_2.10.1     webshot_0.5.5           
##  [35] GenomicAlignments_1.38.0 jsonlite_1.8.7          
##  [37] systemfonts_1.0.5        tools_4.3.2             
##  [39] progress_1.2.2           glue_1.6.2              
##  [41] SparseArray_1.2.2        xfun_0.41               
##  [43] withr_2.5.2              numDeriv_2016.8-1.1     
##  [45] formatR_1.14             fastmap_1.1.1           
##  [47] boot_1.3-28.1            fansi_1.0.5             
##  [49] digest_0.6.33            timechange_0.2.0        
##  [51] R6_2.5.1                 seqPattern_1.34.0       
##  [53] colorspace_2.1-0         biomaRt_2.58.2          
##  [55] RSQLite_2.3.3            utf8_1.2.4              
##  [57] generics_0.1.3           data.table_1.14.8       
##  [59] rtracklayer_1.62.0       prettyunits_1.2.0       
##  [61] httr_1.4.7               S4Arrays_1.2.0          
##  [63] pkgconfig_2.0.3          gtable_0.3.4            
##  [65] blob_1.2.4               impute_1.76.0           
##  [67] XVector_0.42.0           htmltools_0.5.7         
##  [69] bookdown_0.38            scales_1.2.1            
##  [71] png_0.1-8                knitr_1.45              
##  [73] lambda.r_1.2.4           rstudioapi_0.15.0       
##  [75] tzdb_0.4.0               reshape2_1.4.4          
##  [77] rjson_0.2.21             nlme_3.1-163            
##  [79] curl_5.1.0               nloptr_2.0.3            
##  [81] cachem_1.0.8             KernSmooth_2.23-22      
##  [83] restfulr_0.0.15          pillar_1.9.0            
##  [85] vctrs_0.6.4              dbplyr_2.4.0            
##  [87] evaluate_0.23            cli_3.6.1               
##  [89] locfit_1.5-9.9           compiler_4.3.2          
##  [91] futile.options_1.0.1     Rsamtools_2.18.0        
##  [93] rlang_1.1.2              crayon_1.5.2            
##  [95] ggsignif_0.6.4           labeling_0.4.3          
##  [97] stringi_1.8.2            viridisLite_0.4.2       
##  [99] gridBase_0.4-7           BiocParallel_1.36.0     
## [101] munsell_0.5.0            Biostrings_2.70.1       
## [103] BSgenome_1.70.2          hms_1.1.3               
## [105] bit64_4.0.5              KEGGREST_1.42.0         
## [107] highr_0.10               broom_1.0.5             
## [109] memoise_2.0.1            bslib_0.6.0             
## [111] bit_4.0.5</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="sequence-reads-to-snp-level-parent-specific-read-counts-1.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/YOUR GITHUB USERNAME/YOUR REPO NAME/edit/main/02-Part2.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/YOUR GITHUB USERNAME/YOUR REPO NAME/blob/main/02-Part2.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
